// nuxi@3.33.0/dist/build-0VWSSMdh.mjs
import { t as defineCommand } from "./dist-sJJifug_.mjs";
import { n as colors } from "./consola.DXBYu-KD-BxkQ5_xB.mjs";
import "./utils-D5jIbZ8_.mjs";
import { a as legacyRootDirArgs, i as extendsArgs, n as dotEnvArgs, o as logLevelArgs, r as envNameArgs, t as cwdArgs } from "./_shared-CoPxtaYg.mjs";
import { d as Wt, n as logger, s as Nt } from "./logger-C9cMUJ6F.mjs";
import { c as resolve, s as relative } from "./pathe.M-eThtNZ-0R4k6vpe.mjs";
import { t as overrideEnv } from "./env-DPrym_Ws.mjs";
import { t as loadKit } from "./kit-BqwTIe0r.mjs";
import "./satisfies-B5PBn7KX.mjs";
import "./versions-B7j3MyFW.mjs";
import { n as showVersions } from "./banner-B_0Op0tw.mjs";
import { t as clearBuildDir } from "./fs-BgbDM8Fv.mjs";
import process from "node:process";

//#region src/commands/build.ts
var build_default = defineCommand({
	meta: {
		name: "build",
		description: "Build Nuxt for production deployment"
	},
	args: {
		...cwdArgs,
		...logLevelArgs,
		prerender: {
			type: "boolean",
			description: "Build Nuxt and prerender static routes"
		},
		preset: {
			type: "string",
			description: "Nitro server preset"
		},
		...dotEnvArgs,
		...envNameArgs,
		...extendsArgs,
		...legacyRootDirArgs
	},
	async run(ctx) {
		overrideEnv("production");
		const cwd = resolve(ctx.args.cwd || ctx.args.rootDir);
		Nt(colors.cyan("Building Nuxt for production..."));
		const kit = await loadKit(cwd);
		await showVersions(cwd, kit, ctx.args.dotenv);
		const nuxt = await kit.loadNuxt({
			cwd,
			dotenv: {
				cwd,
				fileName: ctx.args.dotenv
			},
			envName: ctx.args.envName,
			overrides: {
				logLevel: ctx.args.logLevel,
				_generate: ctx.args.prerender,
				nitro: {
					static: ctx.args.prerender,
					preset: ctx.args.preset || process.env.NITRO_PRESET || process.env.SERVER_PRESET
				},
				...ctx.args.extends && { extends: ctx.args.extends },
				...ctx.data?.overrides
			}
		});
		let nitro;
		try {
			nitro = kit.useNitro?.();
			if (nitro) logger.info(`Nitro preset: ${colors.cyan(nitro.options.preset)}`);
		} catch {}
		await clearBuildDir(nuxt.options.buildDir);
		await kit.writeTypes(nuxt);
		nuxt.hook("build:error", (err) => {
			logger.error(`Nuxt build error: ${err}`);
			process.exit(1);
		});
		await kit.buildNuxt(nuxt);
		if (ctx.args.prerender) {
			if (!nuxt.options.ssr) {
				logger.warn(`HTML content not prerendered because ${colors.cyan("ssr: false")} was set.`);
				logger.info(`You can read more in ${colors.cyan("https://nuxt.com/docs/getting-started/deployment#static-hosting")}.`);
			}
			const dir = nitro?.options.output.publicDir;
			const publicDir = dir ? relative(process.cwd(), dir) : ".output/public";
			Wt(`✨ You can now deploy ${colors.cyan(publicDir)} to any static hosting!`);
		} else Wt("✨ Build complete!");
	}
});

//#endregion
export { build_default as default };
