// nuxi@3.33.0/dist/generate-ey_Z99Q1.mjs
import { t as defineCommand } from "./dist-sJJifug_.mjs";
import "./consola.DXBYu-KD-BxkQ5_xB.mjs";
import "./utils-D5jIbZ8_.mjs";
import { a as legacyRootDirArgs, i as extendsArgs, n as dotEnvArgs, o as logLevelArgs, r as envNameArgs, t as cwdArgs } from "./_shared-CoPxtaYg.mjs";
import "./logger-C9cMUJ6F.mjs";
import "./env-DPrym_Ws.mjs";
import "./kit-BqwTIe0r.mjs";
import "./satisfies-B5PBn7KX.mjs";
import "./versions-B7j3MyFW.mjs";
import "./banner-B_0Op0tw.mjs";
import "./fs-BgbDM8Fv.mjs";
import build_default from "./build-0VWSSMdh.mjs";

//#region src/commands/generate.ts
var generate_default = defineCommand({
	meta: {
		name: "generate",
		description: "Build Nuxt and prerender all routes"
	},
	args: {
		...cwdArgs,
		...logLevelArgs,
		preset: {
			type: "string",
			description: "Nitro server preset"
		},
		...dotEnvArgs,
		...envNameArgs,
		...extendsArgs,
		...legacyRootDirArgs
	},
	async run(ctx) {
		ctx.args.prerender = true;
		await build_default.run(ctx);
	}
});

//#endregion
export { generate_default as default };
