// nuxi@3.33.0/dist/logger-C9cMUJ6F.mjs
import { a as __toCommonJS, i as __require, n as __esmMin, o as __toESM, r as __exportAll, t as __commonJSMin } from "./chunk-DPkJJFeX.mjs";
import { dirname, join } from "node:path";
import process$1, { stdin, stdout } from "node:process";
import { stripVTControlCharacters } from "node:util";
import tty, { ReadStream } from "node:tty";
import * as k$1 from "node:readline";
import c from "node:readline";
import { existsSync, lstatSync, readdirSync } from "node:fs";
import os from "node:os";

//#region ../../node_modules/.pnpm/picocolors@1.1.1/node_modules/picocolors/picocolors.js
var require_picocolors = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	let p = process || {}, argv = p.argv || [], env = p.env || {};
	let isColorSupported = !(!!env.NO_COLOR || argv.includes("--no-color")) && (!!env.FORCE_COLOR || argv.includes("--color") || p.platform === "win32" || (p.stdout || {}).isTTY && env.TERM !== "dumb" || !!env.CI);
	let formatter = (open, close, replace = open) => (input) => {
		let string = "" + input, index = string.indexOf(close, open.length);
		return ~index ? open + replaceClose(string, close, replace, index) + close : open + string + close;
	};
	let replaceClose = (string, close, replace, index) => {
		let result = "", cursor = 0;
		do {
			result += string.substring(cursor, index) + replace;
			cursor = index + close.length;
			index = string.indexOf(close, cursor);
		} while (~index);
		return result + string.substring(cursor);
	};
	let createColors = (enabled = isColorSupported) => {
		let f = enabled ? formatter : () => String;
		return {
			isColorSupported: enabled,
			reset: f("\x1B[0m", "\x1B[0m"),
			bold: f("\x1B[1m", "\x1B[22m", "\x1B[22m\x1B[1m"),
			dim: f("\x1B[2m", "\x1B[22m", "\x1B[22m\x1B[2m"),
			italic: f("\x1B[3m", "\x1B[23m"),
			underline: f("\x1B[4m", "\x1B[24m"),
			inverse: f("\x1B[7m", "\x1B[27m"),
			hidden: f("\x1B[8m", "\x1B[28m"),
			strikethrough: f("\x1B[9m", "\x1B[29m"),
			black: f("\x1B[30m", "\x1B[39m"),
			red: f("\x1B[31m", "\x1B[39m"),
			green: f("\x1B[32m", "\x1B[39m"),
			yellow: f("\x1B[33m", "\x1B[39m"),
			blue: f("\x1B[34m", "\x1B[39m"),
			magenta: f("\x1B[35m", "\x1B[39m"),
			cyan: f("\x1B[36m", "\x1B[39m"),
			white: f("\x1B[37m", "\x1B[39m"),
			gray: f("\x1B[90m", "\x1B[39m"),
			bgBlack: f("\x1B[40m", "\x1B[49m"),
			bgRed: f("\x1B[41m", "\x1B[49m"),
			bgGreen: f("\x1B[42m", "\x1B[49m"),
			bgYellow: f("\x1B[43m", "\x1B[49m"),
			bgBlue: f("\x1B[44m", "\x1B[49m"),
			bgMagenta: f("\x1B[45m", "\x1B[49m"),
			bgCyan: f("\x1B[46m", "\x1B[49m"),
			bgWhite: f("\x1B[47m", "\x1B[49m"),
			blackBright: f("\x1B[90m", "\x1B[39m"),
			redBright: f("\x1B[91m", "\x1B[39m"),
			greenBright: f("\x1B[92m", "\x1B[39m"),
			yellowBright: f("\x1B[93m", "\x1B[39m"),
			blueBright: f("\x1B[94m", "\x1B[39m"),
			magentaBright: f("\x1B[95m", "\x1B[39m"),
			cyanBright: f("\x1B[96m", "\x1B[39m"),
			whiteBright: f("\x1B[97m", "\x1B[39m"),
			bgBlackBright: f("\x1B[100m", "\x1B[49m"),
			bgRedBright: f("\x1B[101m", "\x1B[49m"),
			bgGreenBright: f("\x1B[102m", "\x1B[49m"),
			bgYellowBright: f("\x1B[103m", "\x1B[49m"),
			bgBlueBright: f("\x1B[104m", "\x1B[49m"),
			bgMagentaBright: f("\x1B[105m", "\x1B[49m"),
			bgCyanBright: f("\x1B[106m", "\x1B[49m"),
			bgWhiteBright: f("\x1B[107m", "\x1B[49m")
		};
	};
	module.exports = createColors();
	module.exports.createColors = createColors;
}));

//#endregion
//#region ../../node_modules/.pnpm/sisteransi@1.0.5/node_modules/sisteransi/src/index.js
var require_src$1 = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	const ESC = "\x1B";
	const CSI = `${ESC}[`;
	const beep = "\x07";
	const cursor = {
		to(x, y) {
			if (!y) return `${CSI}${x + 1}G`;
			return `${CSI}${y + 1};${x + 1}H`;
		},
		move(x, y) {
			let ret = "";
			if (x < 0) ret += `${CSI}${-x}D`;
			else if (x > 0) ret += `${CSI}${x}C`;
			if (y < 0) ret += `${CSI}${-y}A`;
			else if (y > 0) ret += `${CSI}${y}B`;
			return ret;
		},
		up: (count = 1) => `${CSI}${count}A`,
		down: (count = 1) => `${CSI}${count}B`,
		forward: (count = 1) => `${CSI}${count}C`,
		backward: (count = 1) => `${CSI}${count}D`,
		nextLine: (count = 1) => `${CSI}E`.repeat(count),
		prevLine: (count = 1) => `${CSI}F`.repeat(count),
		left: `${CSI}G`,
		hide: `${CSI}?25l`,
		show: `${CSI}?25h`,
		save: `${ESC}7`,
		restore: `${ESC}8`
	};
	const scroll = {
		up: (count = 1) => `${CSI}S`.repeat(count),
		down: (count = 1) => `${CSI}T`.repeat(count)
	};
	const erase = {
		screen: `${CSI}2J`,
		up: (count = 1) => `${CSI}1J`.repeat(count),
		down: (count = 1) => `${CSI}J`.repeat(count),
		line: `${CSI}2K`,
		lineEnd: `${CSI}K`,
		lineStart: `${CSI}1K`,
		lines(count) {
			let clear = "";
			for (let i = 0; i < count; i++) clear += this.line + (i < count - 1 ? cursor.up() : "");
			if (count) clear += cursor.left;
			return clear;
		}
	};
	module.exports = {
		cursor,
		scroll,
		erase,
		beep
	};
}));

//#endregion
//#region ../../node_modules/.pnpm/@clack+core@1.0.0/node_modules/@clack/core/dist/index.mjs
var import_picocolors = /* @__PURE__ */ __toESM(require_picocolors(), 1);
var import_src$1 = require_src$1();
const at = (t) => t === 161 || t === 164 || t === 167 || t === 168 || t === 170 || t === 173 || t === 174 || t >= 176 && t <= 180 || t >= 182 && t <= 186 || t >= 188 && t <= 191 || t === 198 || t === 208 || t === 215 || t === 216 || t >= 222 && t <= 225 || t === 230 || t >= 232 && t <= 234 || t === 236 || t === 237 || t === 240 || t === 242 || t === 243 || t >= 247 && t <= 250 || t === 252 || t === 254 || t === 257 || t === 273 || t === 275 || t === 283 || t === 294 || t === 295 || t === 299 || t >= 305 && t <= 307 || t === 312 || t >= 319 && t <= 322 || t === 324 || t >= 328 && t <= 331 || t === 333 || t === 338 || t === 339 || t === 358 || t === 359 || t === 363 || t === 462 || t === 464 || t === 466 || t === 468 || t === 470 || t === 472 || t === 474 || t === 476 || t === 593 || t === 609 || t === 708 || t === 711 || t >= 713 && t <= 715 || t === 717 || t === 720 || t >= 728 && t <= 731 || t === 733 || t === 735 || t >= 768 && t <= 879 || t >= 913 && t <= 929 || t >= 931 && t <= 937 || t >= 945 && t <= 961 || t >= 963 && t <= 969 || t === 1025 || t >= 1040 && t <= 1103 || t === 1105 || t === 8208 || t >= 8211 && t <= 8214 || t === 8216 || t === 8217 || t === 8220 || t === 8221 || t >= 8224 && t <= 8226 || t >= 8228 && t <= 8231 || t === 8240 || t === 8242 || t === 8243 || t === 8245 || t === 8251 || t === 8254 || t === 8308 || t === 8319 || t >= 8321 && t <= 8324 || t === 8364 || t === 8451 || t === 8453 || t === 8457 || t === 8467 || t === 8470 || t === 8481 || t === 8482 || t === 8486 || t === 8491 || t === 8531 || t === 8532 || t >= 8539 && t <= 8542 || t >= 8544 && t <= 8555 || t >= 8560 && t <= 8569 || t === 8585 || t >= 8592 && t <= 8601 || t === 8632 || t === 8633 || t === 8658 || t === 8660 || t === 8679 || t === 8704 || t === 8706 || t === 8707 || t === 8711 || t === 8712 || t === 8715 || t === 8719 || t === 8721 || t === 8725 || t === 8730 || t >= 8733 && t <= 8736 || t === 8739 || t === 8741 || t >= 8743 && t <= 8748 || t === 8750 || t >= 8756 && t <= 8759 || t === 8764 || t === 8765 || t === 8776 || t === 8780 || t === 8786 || t === 8800 || t === 8801 || t >= 8804 && t <= 8807 || t === 8810 || t === 8811 || t === 8814 || t === 8815 || t === 8834 || t === 8835 || t === 8838 || t === 8839 || t === 8853 || t === 8857 || t === 8869 || t === 8895 || t === 8978 || t >= 9312 && t <= 9449 || t >= 9451 && t <= 9547 || t >= 9552 && t <= 9587 || t >= 9600 && t <= 9615 || t >= 9618 && t <= 9621 || t === 9632 || t === 9633 || t >= 9635 && t <= 9641 || t === 9650 || t === 9651 || t === 9654 || t === 9655 || t === 9660 || t === 9661 || t === 9664 || t === 9665 || t >= 9670 && t <= 9672 || t === 9675 || t >= 9678 && t <= 9681 || t >= 9698 && t <= 9701 || t === 9711 || t === 9733 || t === 9734 || t === 9737 || t === 9742 || t === 9743 || t === 9756 || t === 9758 || t === 9792 || t === 9794 || t === 9824 || t === 9825 || t >= 9827 && t <= 9829 || t >= 9831 && t <= 9834 || t === 9836 || t === 9837 || t === 9839 || t === 9886 || t === 9887 || t === 9919 || t >= 9926 && t <= 9933 || t >= 9935 && t <= 9939 || t >= 9941 && t <= 9953 || t === 9955 || t === 9960 || t === 9961 || t >= 9963 && t <= 9969 || t === 9972 || t >= 9974 && t <= 9977 || t === 9979 || t === 9980 || t === 9982 || t === 9983 || t === 10045 || t >= 10102 && t <= 10111 || t >= 11094 && t <= 11097 || t >= 12872 && t <= 12879 || t >= 57344 && t <= 63743 || t >= 65024 && t <= 65039 || t === 65533 || t >= 127232 && t <= 127242 || t >= 127248 && t <= 127277 || t >= 127280 && t <= 127337 || t >= 127344 && t <= 127373 || t === 127375 || t === 127376 || t >= 127387 && t <= 127404 || t >= 917760 && t <= 917999 || t >= 983040 && t <= 1048573 || t >= 1048576 && t <= 1114109, lt = (t) => t === 12288 || t >= 65281 && t <= 65376 || t >= 65504 && t <= 65510, ht$1 = (t) => t >= 4352 && t <= 4447 || t === 8986 || t === 8987 || t === 9001 || t === 9002 || t >= 9193 && t <= 9196 || t === 9200 || t === 9203 || t === 9725 || t === 9726 || t === 9748 || t === 9749 || t >= 9800 && t <= 9811 || t === 9855 || t === 9875 || t === 9889 || t === 9898 || t === 9899 || t === 9917 || t === 9918 || t === 9924 || t === 9925 || t === 9934 || t === 9940 || t === 9962 || t === 9970 || t === 9971 || t === 9973 || t === 9978 || t === 9981 || t === 9989 || t === 9994 || t === 9995 || t === 10024 || t === 10060 || t === 10062 || t >= 10067 && t <= 10069 || t === 10071 || t >= 10133 && t <= 10135 || t === 10160 || t === 10175 || t === 11035 || t === 11036 || t === 11088 || t === 11093 || t >= 11904 && t <= 11929 || t >= 11931 && t <= 12019 || t >= 12032 && t <= 12245 || t >= 12272 && t <= 12287 || t >= 12289 && t <= 12350 || t >= 12353 && t <= 12438 || t >= 12441 && t <= 12543 || t >= 12549 && t <= 12591 || t >= 12593 && t <= 12686 || t >= 12688 && t <= 12771 || t >= 12783 && t <= 12830 || t >= 12832 && t <= 12871 || t >= 12880 && t <= 19903 || t >= 19968 && t <= 42124 || t >= 42128 && t <= 42182 || t >= 43360 && t <= 43388 || t >= 44032 && t <= 55203 || t >= 63744 && t <= 64255 || t >= 65040 && t <= 65049 || t >= 65072 && t <= 65106 || t >= 65108 && t <= 65126 || t >= 65128 && t <= 65131 || t >= 94176 && t <= 94180 || t === 94192 || t === 94193 || t >= 94208 && t <= 100343 || t >= 100352 && t <= 101589 || t >= 101632 && t <= 101640 || t >= 110576 && t <= 110579 || t >= 110581 && t <= 110587 || t === 110589 || t === 110590 || t >= 110592 && t <= 110882 || t === 110898 || t >= 110928 && t <= 110930 || t === 110933 || t >= 110948 && t <= 110951 || t >= 110960 && t <= 111355 || t === 126980 || t === 127183 || t === 127374 || t >= 127377 && t <= 127386 || t >= 127488 && t <= 127490 || t >= 127504 && t <= 127547 || t >= 127552 && t <= 127560 || t === 127568 || t === 127569 || t >= 127584 && t <= 127589 || t >= 127744 && t <= 127776 || t >= 127789 && t <= 127797 || t >= 127799 && t <= 127868 || t >= 127870 && t <= 127891 || t >= 127904 && t <= 127946 || t >= 127951 && t <= 127955 || t >= 127968 && t <= 127984 || t === 127988 || t >= 127992 && t <= 128062 || t === 128064 || t >= 128066 && t <= 128252 || t >= 128255 && t <= 128317 || t >= 128331 && t <= 128334 || t >= 128336 && t <= 128359 || t === 128378 || t === 128405 || t === 128406 || t === 128420 || t >= 128507 && t <= 128591 || t >= 128640 && t <= 128709 || t === 128716 || t >= 128720 && t <= 128722 || t >= 128725 && t <= 128727 || t >= 128732 && t <= 128735 || t === 128747 || t === 128748 || t >= 128756 && t <= 128764 || t >= 128992 && t <= 129003 || t === 129008 || t >= 129292 && t <= 129338 || t >= 129340 && t <= 129349 || t >= 129351 && t <= 129535 || t >= 129648 && t <= 129660 || t >= 129664 && t <= 129672 || t >= 129680 && t <= 129725 || t >= 129727 && t <= 129733 || t >= 129742 && t <= 129755 || t >= 129760 && t <= 129768 || t >= 129776 && t <= 129784 || t >= 131072 && t <= 196605 || t >= 196608 && t <= 262141, O = /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/y, y = /[\x00-\x08\x0A-\x1F\x7F-\x9F]{1,1000}/y, M$1 = /\t{1,1000}/y, P = /[\u{1F1E6}-\u{1F1FF}]{2}|\u{1F3F4}[\u{E0061}-\u{E007A}]{2}[\u{E0030}-\u{E0039}\u{E0061}-\u{E007A}]{1,3}\u{E007F}|(?:\p{Emoji}\uFE0F\u20E3?|\p{Emoji_Modifier_Base}\p{Emoji_Modifier}?|\p{Emoji_Presentation})(?:\u200D(?:\p{Emoji_Modifier_Base}\p{Emoji_Modifier}?|\p{Emoji_Presentation}|\p{Emoji}\uFE0F\u20E3?))*/uy, L = /(?:[\x20-\x7E\xA0-\xFF](?!\uFE0F)){1,1000}/y, ct = /\p{M}+/gu, pt$1 = {
	limit: Infinity,
	ellipsis: ""
}, X = (t, e = {}, s = {}) => {
	const i = e.limit ?? Infinity, r = e.ellipsis ?? "", n = e?.ellipsisWidth ?? (r ? X(r, pt$1, s).width : 0), u = s.ansiWidth ?? 0, a = s.controlWidth ?? 0, l = s.tabWidth ?? 8, E = s.ambiguousWidth ?? 1, g = s.emojiWidth ?? 2, m = s.fullWidthWidth ?? 2, A = s.regularWidth ?? 1, V = s.wideWidth ?? 2;
	let h = 0, o = 0, f = t.length, v = 0, F = !1, d = f, b = Math.max(0, i - n), C = 0, B = 0, c = 0, p = 0;
	t: for (;;) {
		if (B > C || o >= f && o > h) {
			const ut = t.slice(C, B) || t.slice(h, o);
			v = 0;
			for (const Y of ut.replaceAll(ct, "")) {
				const $ = Y.codePointAt(0) || 0;
				if (lt($) ? p = m : ht$1($) ? p = V : E !== A && at($) ? p = E : p = A, c + p > b && (d = Math.min(d, Math.max(C, h) + v)), c + p > i) {
					F = !0;
					break t;
				}
				v += Y.length, c += p;
			}
			C = B = 0;
		}
		if (o >= f) break;
		if (L.lastIndex = o, L.test(t)) {
			if (v = L.lastIndex - o, p = v * A, c + p > b && (d = Math.min(d, o + Math.floor((b - c) / A))), c + p > i) {
				F = !0;
				break;
			}
			c += p, C = h, B = o, o = h = L.lastIndex;
			continue;
		}
		if (O.lastIndex = o, O.test(t)) {
			if (c + u > b && (d = Math.min(d, o)), c + u > i) {
				F = !0;
				break;
			}
			c += u, C = h, B = o, o = h = O.lastIndex;
			continue;
		}
		if (y.lastIndex = o, y.test(t)) {
			if (v = y.lastIndex - o, p = v * a, c + p > b && (d = Math.min(d, o + Math.floor((b - c) / a))), c + p > i) {
				F = !0;
				break;
			}
			c += p, C = h, B = o, o = h = y.lastIndex;
			continue;
		}
		if (M$1.lastIndex = o, M$1.test(t)) {
			if (v = M$1.lastIndex - o, p = v * l, c + p > b && (d = Math.min(d, o + Math.floor((b - c) / l))), c + p > i) {
				F = !0;
				break;
			}
			c += p, C = h, B = o, o = h = M$1.lastIndex;
			continue;
		}
		if (P.lastIndex = o, P.test(t)) {
			if (c + g > b && (d = Math.min(d, o)), c + g > i) {
				F = !0;
				break;
			}
			c += g, C = h, B = o, o = h = P.lastIndex;
			continue;
		}
		o += 1;
	}
	return {
		width: F ? b : c,
		index: F ? d : f,
		truncated: F,
		ellipsed: F && i >= n
	};
}, ft$1 = {
	limit: Infinity,
	ellipsis: "",
	ellipsisWidth: 0
}, S = (t, e = {}) => X(t, ft$1, e).width, W = "\x1B", Z = "", Ft$1 = 39, j = "\x07", Q$1 = "[", dt = "]", tt = "m", U = `${dt}8;;`, et = new RegExp(`(?:\\${Q$1}(?<code>\\d+)m|\\${U}(?<uri>.*)${j})`, "y"), mt$1 = (t) => {
	if (t >= 30 && t <= 37 || t >= 90 && t <= 97) return 39;
	if (t >= 40 && t <= 47 || t >= 100 && t <= 107) return 49;
	if (t === 1 || t === 2) return 22;
	if (t === 3) return 23;
	if (t === 4) return 24;
	if (t === 7) return 27;
	if (t === 8) return 28;
	if (t === 9) return 29;
	if (t === 0) return 0;
}, st = (t) => `${W}${Q$1}${t}${tt}`, it = (t) => `${W}${U}${t}${j}`, gt$1 = (t) => t.map((e) => S(e)), G$1 = (t, e, s) => {
	const i = e[Symbol.iterator]();
	let r = !1, n = !1, u = t.at(-1), a = u === void 0 ? 0 : S(u), l = i.next(), E = i.next(), g = 0;
	for (; !l.done;) {
		const m = l.value, A = S(m);
		a + A <= s ? t[t.length - 1] += m : (t.push(m), a = 0), (m === W || m === Z) && (r = !0, n = e.startsWith(U, g + 1)), r ? n ? m === j && (r = !1, n = !1) : m === tt && (r = !1) : (a += A, a === s && !E.done && (t.push(""), a = 0)), l = E, E = i.next(), g += m.length;
	}
	u = t.at(-1), !a && u !== void 0 && u.length > 0 && t.length > 1 && (t[t.length - 2] += t.pop());
}, vt$1 = (t) => {
	const e = t.split(" ");
	let s = e.length;
	for (; s > 0 && !(S(e[s - 1]) > 0);) s--;
	return s === e.length ? t : e.slice(0, s).join(" ") + e.slice(s).join("");
}, Et$1 = (t, e, s = {}) => {
	if (s.trim !== !1 && t.trim() === "") return "";
	let i = "", r, n;
	const u = t.split(" "), a = gt$1(u);
	let l = [""];
	for (const [h, o] of u.entries()) {
		s.trim !== !1 && (l[l.length - 1] = (l.at(-1) ?? "").trimStart());
		let f = S(l.at(-1) ?? "");
		if (h !== 0 && (f >= e && (s.wordWrap === !1 || s.trim === !1) && (l.push(""), f = 0), (f > 0 || s.trim === !1) && (l[l.length - 1] += " ", f++)), s.hard && a[h] > e) {
			const v = e - f, F = 1 + Math.floor((a[h] - v - 1) / e);
			Math.floor((a[h] - 1) / e) < F && l.push(""), G$1(l, o, e);
			continue;
		}
		if (f + a[h] > e && f > 0 && a[h] > 0) {
			if (s.wordWrap === !1 && f < e) {
				G$1(l, o, e);
				continue;
			}
			l.push("");
		}
		if (f + a[h] > e && s.wordWrap === !1) {
			G$1(l, o, e);
			continue;
		}
		l[l.length - 1] += o;
	}
	s.trim !== !1 && (l = l.map((h) => vt$1(h)));
	const E = l.join(`
`), g = E[Symbol.iterator]();
	let m = g.next(), A = g.next(), V = 0;
	for (; !m.done;) {
		const h = m.value, o = A.value;
		if (i += h, h === W || h === Z) {
			et.lastIndex = V + 1;
			const F = et.exec(E)?.groups;
			if (F?.code !== void 0) {
				const d = Number.parseFloat(F.code);
				r = d === Ft$1 ? void 0 : d;
			} else F?.uri !== void 0 && (n = F.uri.length === 0 ? void 0 : F.uri);
		}
		const f = r ? mt$1(r) : void 0;
		o === `
` ? (n && (i += it("")), r && f && (i += st(f))) : h === `
` && (r && f && (i += st(r)), n && (i += it(n))), V += h.length, m = A, A = g.next();
	}
	return i;
};
function K$1(t, e, s) {
	return String(t).normalize().replaceAll(`\r
`, `
`).split(`
`).map((i) => Et$1(i, e, s)).join(`
`);
}
const _ = {
	actions: new Set([
		"up",
		"down",
		"left",
		"right",
		"space",
		"enter",
		"cancel"
	]),
	aliases: new Map([
		["k", "up"],
		["j", "down"],
		["h", "left"],
		["l", "right"],
		["", "cancel"],
		["escape", "cancel"]
	]),
	messages: {
		cancel: "Canceled",
		error: "Something went wrong"
	},
	withGuide: !0
};
function H$1(t, e) {
	if (typeof t == "string") return _.aliases.get(t) === e;
	for (const s of t) if (s !== void 0 && H$1(s, e)) return !0;
	return !1;
}
function _t$1(t, e) {
	if (t === e) return;
	const s = t.split(`
`), i = e.split(`
`), r = Math.max(s.length, i.length), n = [];
	for (let u = 0; u < r; u++) s[u] !== i[u] && n.push(u);
	return {
		lines: n,
		numLinesBefore: s.length,
		numLinesAfter: i.length,
		numLines: r
	};
}
const bt$1 = globalThis.process.platform.startsWith("win"), z$1 = Symbol("clack:cancel");
function Ct$1(t) {
	return t === z$1;
}
function T(t, e) {
	const s = t;
	s.isTTY && s.setRawMode(e);
}
function xt$1({ input: t = stdin, output: e = stdout, overwrite: s = !0, hideCursor: i = !0 } = {}) {
	const r = k$1.createInterface({
		input: t,
		output: e,
		prompt: "",
		tabSize: 1
	});
	k$1.emitKeypressEvents(t, r), t instanceof ReadStream && t.isTTY && t.setRawMode(!0);
	const n = (u, { name: a, sequence: l }) => {
		if (H$1([
			String(u),
			a,
			l
		], "cancel")) {
			i && e.write(import_src$1.cursor.show), process.exit(0);
			return;
		}
		if (!s) return;
		const g = a === "return" ? 0 : -1, m = a === "return" ? -1 : 0;
		k$1.moveCursor(e, g, m, () => {
			k$1.clearLine(e, 1, () => {
				t.once("keypress", n);
			});
		});
	};
	return i && e.write(import_src$1.cursor.hide), t.once("keypress", n), () => {
		t.off("keypress", n), i && e.write(import_src$1.cursor.show), t instanceof ReadStream && t.isTTY && !bt$1 && t.setRawMode(!1), r.terminal = !1, r.close();
	};
}
const rt = (t) => "columns" in t && typeof t.columns == "number" ? t.columns : 80, nt = (t) => "rows" in t && typeof t.rows == "number" ? t.rows : 20;
function Bt$1(t, e, s, i = s) {
	return K$1(e, rt(t ?? stdout) - s.length, {
		hard: !0,
		trim: !1
	}).split(`
`).map((n, u) => `${u === 0 ? i : s}${n}`).join(`
`);
}
var x$1 = class {
	input;
	output;
	_abortSignal;
	rl;
	opts;
	_render;
	_track = !1;
	_prevFrame = "";
	_subscribers = /* @__PURE__ */ new Map();
	_cursor = 0;
	state = "initial";
	error = "";
	value;
	userInput = "";
	constructor(e, s = !0) {
		const { input: i = stdin, output: r = stdout, render: n, signal: u, ...a } = e;
		this.opts = a, this.onKeypress = this.onKeypress.bind(this), this.close = this.close.bind(this), this.render = this.render.bind(this), this._render = n.bind(this), this._track = s, this._abortSignal = u, this.input = i, this.output = r;
	}
	unsubscribe() {
		this._subscribers.clear();
	}
	setSubscriber(e, s) {
		const i = this._subscribers.get(e) ?? [];
		i.push(s), this._subscribers.set(e, i);
	}
	on(e, s) {
		this.setSubscriber(e, { cb: s });
	}
	once(e, s) {
		this.setSubscriber(e, {
			cb: s,
			once: !0
		});
	}
	emit(e, ...s) {
		const i = this._subscribers.get(e) ?? [], r = [];
		for (const n of i) n.cb(...s), n.once && r.push(() => i.splice(i.indexOf(n), 1));
		for (const n of r) n();
	}
	prompt() {
		return new Promise((e) => {
			if (this._abortSignal) {
				if (this._abortSignal.aborted) return this.state = "cancel", this.close(), e(z$1);
				this._abortSignal.addEventListener("abort", () => {
					this.state = "cancel", this.close();
				}, { once: !0 });
			}
			this.rl = c.createInterface({
				input: this.input,
				tabSize: 2,
				prompt: "",
				escapeCodeTimeout: 50,
				terminal: !0
			}), this.rl.prompt(), this.opts.initialUserInput !== void 0 && this._setUserInput(this.opts.initialUserInput, !0), this.input.on("keypress", this.onKeypress), T(this.input, !0), this.output.on("resize", this.render), this.render(), this.once("submit", () => {
				this.output.write(import_src$1.cursor.show), this.output.off("resize", this.render), T(this.input, !1), e(this.value);
			}), this.once("cancel", () => {
				this.output.write(import_src$1.cursor.show), this.output.off("resize", this.render), T(this.input, !1), e(z$1);
			});
		});
	}
	_isActionKey(e, s) {
		return e === "	";
	}
	_setValue(e) {
		this.value = e, this.emit("value", this.value);
	}
	_setUserInput(e, s) {
		this.userInput = e ?? "", this.emit("userInput", this.userInput), s && this._track && this.rl && (this.rl.write(this.userInput), this._cursor = this.rl.cursor);
	}
	_clearUserInput() {
		this.rl?.write(null, {
			ctrl: !0,
			name: "u"
		}), this._setUserInput("");
	}
	onKeypress(e, s) {
		if (this._track && s.name !== "return" && (s.name && this._isActionKey(e, s) && this.rl?.write(null, {
			ctrl: !0,
			name: "h"
		}), this._cursor = this.rl?.cursor ?? 0, this._setUserInput(this.rl?.line)), this.state === "error" && (this.state = "active"), s?.name && (!this._track && _.aliases.has(s.name) && this.emit("cursor", _.aliases.get(s.name)), _.actions.has(s.name) && this.emit("cursor", s.name)), e && (e.toLowerCase() === "y" || e.toLowerCase() === "n") && this.emit("confirm", e.toLowerCase() === "y"), this.emit("key", e?.toLowerCase(), s), s?.name === "return") {
			if (this.opts.validate) {
				const i = this.opts.validate(this.value);
				i && (this.error = i instanceof Error ? i.message : i, this.state = "error", this.rl?.write(this.userInput));
			}
			this.state !== "error" && (this.state = "submit");
		}
		H$1([
			e,
			s?.name,
			s?.sequence
		], "cancel") && (this.state = "cancel"), (this.state === "submit" || this.state === "cancel") && this.emit("finalize"), this.render(), (this.state === "submit" || this.state === "cancel") && this.close();
	}
	close() {
		this.input.unpipe(), this.input.removeListener("keypress", this.onKeypress), this.output.write(`
`), T(this.input, !1), this.rl?.close(), this.rl = void 0, this.emit(`${this.state}`, this.value), this.unsubscribe();
	}
	restoreCursor() {
		const e = K$1(this._prevFrame, process.stdout.columns, {
			hard: !0,
			trim: !1
		}).split(`
`).length - 1;
		this.output.write(import_src$1.cursor.move(-999, e * -1));
	}
	render() {
		const e = K$1(this._render(this) ?? "", process.stdout.columns, {
			hard: !0,
			trim: !1
		});
		if (e !== this._prevFrame) {
			if (this.state === "initial") this.output.write(import_src$1.cursor.hide);
			else {
				const s = _t$1(this._prevFrame, e), i = nt(this.output);
				if (this.restoreCursor(), s) {
					const r = Math.max(0, s.numLinesAfter - i), n = Math.max(0, s.numLinesBefore - i);
					let u = s.lines.find((a) => a >= r);
					if (u === void 0) {
						this._prevFrame = e;
						return;
					}
					if (s.lines.length === 1) {
						this.output.write(import_src$1.cursor.move(0, u - n)), this.output.write(import_src$1.erase.lines(1));
						const a = e.split(`
`);
						this.output.write(a[u]), this._prevFrame = e, this.output.write(import_src$1.cursor.move(0, a.length - u - 1));
						return;
					} else if (s.lines.length > 1) {
						if (r < n) u = r;
						else {
							const l = u - n;
							l > 0 && this.output.write(import_src$1.cursor.move(0, l));
						}
						this.output.write(import_src$1.erase.down());
						const a = e.split(`
`).slice(u);
						this.output.write(a.join(`
`)), this._prevFrame = e;
						return;
					}
				}
				this.output.write(import_src$1.erase.down());
			}
			this.output.write(e), this.state === "initial" && (this.state = "active"), this._prevFrame = e;
		}
	}
};
function wt$1(t, e) {
	if (t === void 0 || e.length === 0) return 0;
	const s = e.findIndex((i) => i.value === t);
	return s !== -1 ? s : 0;
}
function Dt$1(t, e) {
	return (e.label ?? String(e.value)).toLowerCase().includes(t.toLowerCase());
}
function St$1(t, e) {
	if (e) return t ? e : e[0];
}
var Vt$1 = class extends x$1 {
	filteredOptions;
	multiple;
	isNavigating = !1;
	selectedValues = [];
	focusedValue;
	#t = 0;
	#s = "";
	#i;
	#e;
	get cursor() {
		return this.#t;
	}
	get userInputWithCursor() {
		if (!this.userInput) return import_picocolors.default.inverse(import_picocolors.default.hidden("_"));
		if (this._cursor >= this.userInput.length) return `${this.userInput}\u2588`;
		const e = this.userInput.slice(0, this._cursor), [s, ...i] = this.userInput.slice(this._cursor);
		return `${e}${import_picocolors.default.inverse(s)}${i.join("")}`;
	}
	get options() {
		return typeof this.#e == "function" ? this.#e() : this.#e;
	}
	constructor(e) {
		super(e), this.#e = e.options;
		const s = this.options;
		this.filteredOptions = [...s], this.multiple = e.multiple === !0, this.#i = e.filter ?? Dt$1;
		let i;
		if (e.initialValue && Array.isArray(e.initialValue) ? this.multiple ? i = e.initialValue : i = e.initialValue.slice(0, 1) : !this.multiple && this.options.length > 0 && (i = [this.options[0].value]), i) for (const r of i) {
			const n = s.findIndex((u) => u.value === r);
			n !== -1 && (this.toggleSelected(r), this.#t = n);
		}
		this.focusedValue = this.options[this.#t]?.value, this.on("key", (r, n) => this.#r(r, n)), this.on("userInput", (r) => this.#n(r));
	}
	_isActionKey(e, s) {
		return e === "	" || this.multiple && this.isNavigating && s.name === "space" && e !== void 0 && e !== "";
	}
	#r(e, s) {
		const i = s.name === "up", r = s.name === "down", n = s.name === "return";
		i || r ? (this.#t = Math.max(0, Math.min(this.#t + (i ? -1 : 1), this.filteredOptions.length - 1)), this.focusedValue = this.filteredOptions[this.#t]?.value, this.multiple || (this.selectedValues = [this.focusedValue]), this.isNavigating = !0) : n ? this.value = St$1(this.multiple, this.selectedValues) : this.multiple ? this.focusedValue !== void 0 && (s.name === "tab" || this.isNavigating && s.name === "space") ? this.toggleSelected(this.focusedValue) : this.isNavigating = !1 : (this.focusedValue && (this.selectedValues = [this.focusedValue]), this.isNavigating = !1);
	}
	deselectAll() {
		this.selectedValues = [];
	}
	toggleSelected(e) {
		this.filteredOptions.length !== 0 && (this.multiple ? this.selectedValues.includes(e) ? this.selectedValues = this.selectedValues.filter((s) => s !== e) : this.selectedValues = [...this.selectedValues, e] : this.selectedValues = [e]);
	}
	#n(e) {
		if (e !== this.#s) {
			this.#s = e;
			const s = this.options;
			e ? this.filteredOptions = s.filter((i) => this.#i(e, i)) : this.filteredOptions = [...s], this.#t = wt$1(this.focusedValue, this.filteredOptions), this.focusedValue = this.filteredOptions[this.#t]?.value, this.multiple || (this.focusedValue !== void 0 ? this.toggleSelected(this.focusedValue) : this.deselectAll());
		}
	}
};
var kt$1 = class extends x$1 {
	get cursor() {
		return this.value ? 0 : 1;
	}
	get _value() {
		return this.cursor === 0;
	}
	constructor(e) {
		super(e, !1), this.value = !!e.initialValue, this.on("userInput", () => {
			this.value = this._value;
		}), this.on("confirm", (s) => {
			this.output.write(import_src$1.cursor.move(0, -1)), this.value = s, this.state = "submit", this.close();
		}), this.on("cursor", () => {
			this.value = !this.value;
		});
	}
};
var yt$1 = class extends x$1 {
	options;
	cursor = 0;
	#t;
	getGroupItems(e) {
		return this.options.filter((s) => s.group === e);
	}
	isGroupSelected(e) {
		const s = this.getGroupItems(e), i = this.value;
		return i === void 0 ? !1 : s.every((r) => i.includes(r.value));
	}
	toggleValue() {
		const e = this.options[this.cursor];
		if (this.value === void 0 && (this.value = []), e.group === !0) {
			const s = e.value, i = this.getGroupItems(s);
			this.isGroupSelected(s) ? this.value = this.value.filter((r) => i.findIndex((n) => n.value === r) === -1) : this.value = [...this.value, ...i.map((r) => r.value)], this.value = Array.from(new Set(this.value));
		} else this.value = this.value.includes(e.value) ? this.value.filter((i) => i !== e.value) : [...this.value, e.value];
	}
	constructor(e) {
		super(e, !1);
		const { options: s } = e;
		this.#t = e.selectableGroups !== !1, this.options = Object.entries(s).flatMap(([i, r]) => [{
			value: i,
			group: !0,
			label: i
		}, ...r.map((n) => ({
			...n,
			group: i
		}))]), this.value = [...e.initialValues ?? []], this.cursor = Math.max(this.options.findIndex(({ value: i }) => i === e.cursorAt), this.#t ? 0 : 1), this.on("cursor", (i) => {
			switch (i) {
				case "left":
				case "up": {
					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
					const r = this.options[this.cursor]?.group === !0;
					!this.#t && r && (this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1);
					break;
				}
				case "down":
				case "right": {
					this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1;
					const r = this.options[this.cursor]?.group === !0;
					!this.#t && r && (this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1);
					break;
				}
				case "space":
					this.toggleValue();
					break;
			}
		});
	}
};
function D(t, e, s) {
	const i = t + e, r = Math.max(s.length - 1, 0), n = i < 0 ? r : i > r ? 0 : i;
	return s[n].disabled ? D(n, e < 0 ? -1 : 1, s) : n;
}
var Mt$1 = class extends x$1 {
	options;
	cursor = 0;
	get _value() {
		return this.options[this.cursor].value;
	}
	get _enabledOptions() {
		return this.options.filter((e) => e.disabled !== !0);
	}
	toggleAll() {
		const e = this._enabledOptions;
		this.value = this.value !== void 0 && this.value.length === e.length ? [] : e.map((i) => i.value);
	}
	toggleInvert() {
		const e = this.value;
		if (!e) return;
		this.value = this._enabledOptions.filter((i) => !e.includes(i.value)).map((i) => i.value);
	}
	toggleValue() {
		this.value === void 0 && (this.value = []);
		this.value = this.value.includes(this._value) ? this.value.filter((s) => s !== this._value) : [...this.value, this._value];
	}
	constructor(e) {
		super(e, !1), this.options = e.options, this.value = [...e.initialValues ?? []];
		const s = Math.max(this.options.findIndex(({ value: i }) => i === e.cursorAt), 0);
		this.cursor = this.options[s].disabled ? D(s, 1, this.options) : s, this.on("key", (i) => {
			i === "a" && this.toggleAll(), i === "i" && this.toggleInvert();
		}), this.on("cursor", (i) => {
			switch (i) {
				case "left":
				case "up":
					this.cursor = D(this.cursor, -1, this.options);
					break;
				case "down":
				case "right":
					this.cursor = D(this.cursor, 1, this.options);
					break;
				case "space":
					this.toggleValue();
					break;
			}
		});
	}
};
let Lt$1 = class extends x$1 {
	_mask = "•";
	get cursor() {
		return this._cursor;
	}
	get masked() {
		return this.userInput.replaceAll(/./g, this._mask);
	}
	get userInputWithCursor() {
		if (this.state === "submit" || this.state === "cancel") return this.masked;
		const e = this.userInput;
		if (this.cursor >= e.length) return `${this.masked}${import_picocolors.default.inverse(import_picocolors.default.hidden("_"))}`;
		const s = this.masked, i = s.slice(0, this.cursor), r = s.slice(this.cursor);
		return `${i}${import_picocolors.default.inverse(r[0])}${r.slice(1)}`;
	}
	clear() {
		this._clearUserInput();
	}
	constructor({ mask: e, ...s }) {
		super(s), this._mask = e ?? "•", this.on("userInput", (i) => {
			this._setValue(i);
		});
	}
};
var Wt$1 = class extends x$1 {
	options;
	cursor = 0;
	get _selectedValue() {
		return this.options[this.cursor];
	}
	changeValue() {
		this.value = this._selectedValue.value;
	}
	constructor(e) {
		super(e, !1), this.options = e.options;
		const s = this.options.findIndex(({ value: r }) => r === e.initialValue), i = s === -1 ? 0 : s;
		this.cursor = this.options[i].disabled ? D(i, 1, this.options) : i, this.changeValue(), this.on("cursor", (r) => {
			switch (r) {
				case "left":
				case "up":
					this.cursor = D(this.cursor, -1, this.options);
					break;
				case "down":
				case "right":
					this.cursor = D(this.cursor, 1, this.options);
					break;
			}
			this.changeValue();
		});
	}
};
var Tt$1 = class extends x$1 {
	options;
	cursor = 0;
	constructor(e) {
		super(e, !1), this.options = e.options;
		const s = e.caseSensitive === !0, i = this.options.map(({ value: [r] }) => s ? r : r?.toLowerCase());
		this.cursor = Math.max(i.indexOf(e.initialValue), 0), this.on("key", (r, n) => {
			if (!r) return;
			const u = s && n.shift ? r.toUpperCase() : r;
			if (!i.includes(u)) return;
			const a = this.options.find(({ value: [l] }) => s ? l === u : l?.toLowerCase() === r);
			a && (this.value = a.value, this.state = "submit", this.emit("submit"));
		});
	}
};
var $t = class extends x$1 {
	get userInputWithCursor() {
		if (this.state === "submit") return this.userInput;
		const e = this.userInput;
		if (this.cursor >= e.length) return `${this.userInput}\u2588`;
		const s = e.slice(0, this.cursor), [i, ...r] = e.slice(this.cursor);
		return `${s}${import_picocolors.default.inverse(i)}${r.join("")}`;
	}
	get cursor() {
		return this._cursor;
	}
	constructor(e) {
		super({
			...e,
			initialUserInput: e.initialUserInput ?? e.initialValue
		}), this.on("userInput", (s) => {
			this._setValue(s);
		}), this.on("finalize", () => {
			this.value || (this.value = e.defaultValue), this.value === void 0 && (this.value = "");
		});
	}
};

//#endregion
//#region ../../node_modules/.pnpm/@clack+prompts@1.0.0/node_modules/@clack/prompts/dist/index.mjs
function ht() {
	return process$1.platform !== "win32" ? process$1.env.TERM !== "linux" : !!process$1.env.CI || !!process$1.env.WT_SESSION || !!process$1.env.TERMINUS_SUBLIME || process$1.env.ConEmuTask === "{cmd::Cmder}" || process$1.env.TERM_PROGRAM === "Terminus-Sublime" || process$1.env.TERM_PROGRAM === "vscode" || process$1.env.TERM === "xterm-256color" || process$1.env.TERM === "alacritty" || process$1.env.TERMINAL_EMULATOR === "JetBrains-JediTerm";
}
const ee = ht(), ue = () => process.env.CI === "true", Te = (e) => e.isTTY === !0, w = (e, r) => ee ? e : r, Me = w("◆", "*"), ce = w("■", "x"), de = w("▲", "x"), k = w("◇", "o"), $e = w("┌", "T"), h = w("│", "|"), x = w("└", "—"), Re = w("┐", "T"), Oe = w("┘", "—"), Y = w("●", ">"), K = w("○", " "), te = w("◻", "[•]"), G = w("◼", "[+]"), z = w("◻", "[ ]"), Pe = w("▪", "•"), se = w("─", "-"), he = w("╮", "+"), Ne = w("├", "+"), me = w("╯", "+"), pe = w("╰", "+"), We = w("╭", "+"), ge = w("●", "•"), fe = w("◆", "*"), Fe = w("▲", "!"), ye = w("■", "x"), N = (e) => {
	switch (e) {
		case "initial":
		case "active": return import_picocolors.default.cyan(Me);
		case "cancel": return import_picocolors.default.red(ce);
		case "error": return import_picocolors.default.yellow(de);
		case "submit": return import_picocolors.default.green(k);
	}
}, Ee = (e) => {
	switch (e) {
		case "initial":
		case "active": return import_picocolors.default.cyan(h);
		case "cancel": return import_picocolors.default.red(h);
		case "error": return import_picocolors.default.yellow(h);
		case "submit": return import_picocolors.default.green(h);
	}
}, mt = (e) => e === 161 || e === 164 || e === 167 || e === 168 || e === 170 || e === 173 || e === 174 || e >= 176 && e <= 180 || e >= 182 && e <= 186 || e >= 188 && e <= 191 || e === 198 || e === 208 || e === 215 || e === 216 || e >= 222 && e <= 225 || e === 230 || e >= 232 && e <= 234 || e === 236 || e === 237 || e === 240 || e === 242 || e === 243 || e >= 247 && e <= 250 || e === 252 || e === 254 || e === 257 || e === 273 || e === 275 || e === 283 || e === 294 || e === 295 || e === 299 || e >= 305 && e <= 307 || e === 312 || e >= 319 && e <= 322 || e === 324 || e >= 328 && e <= 331 || e === 333 || e === 338 || e === 339 || e === 358 || e === 359 || e === 363 || e === 462 || e === 464 || e === 466 || e === 468 || e === 470 || e === 472 || e === 474 || e === 476 || e === 593 || e === 609 || e === 708 || e === 711 || e >= 713 && e <= 715 || e === 717 || e === 720 || e >= 728 && e <= 731 || e === 733 || e === 735 || e >= 768 && e <= 879 || e >= 913 && e <= 929 || e >= 931 && e <= 937 || e >= 945 && e <= 961 || e >= 963 && e <= 969 || e === 1025 || e >= 1040 && e <= 1103 || e === 1105 || e === 8208 || e >= 8211 && e <= 8214 || e === 8216 || e === 8217 || e === 8220 || e === 8221 || e >= 8224 && e <= 8226 || e >= 8228 && e <= 8231 || e === 8240 || e === 8242 || e === 8243 || e === 8245 || e === 8251 || e === 8254 || e === 8308 || e === 8319 || e >= 8321 && e <= 8324 || e === 8364 || e === 8451 || e === 8453 || e === 8457 || e === 8467 || e === 8470 || e === 8481 || e === 8482 || e === 8486 || e === 8491 || e === 8531 || e === 8532 || e >= 8539 && e <= 8542 || e >= 8544 && e <= 8555 || e >= 8560 && e <= 8569 || e === 8585 || e >= 8592 && e <= 8601 || e === 8632 || e === 8633 || e === 8658 || e === 8660 || e === 8679 || e === 8704 || e === 8706 || e === 8707 || e === 8711 || e === 8712 || e === 8715 || e === 8719 || e === 8721 || e === 8725 || e === 8730 || e >= 8733 && e <= 8736 || e === 8739 || e === 8741 || e >= 8743 && e <= 8748 || e === 8750 || e >= 8756 && e <= 8759 || e === 8764 || e === 8765 || e === 8776 || e === 8780 || e === 8786 || e === 8800 || e === 8801 || e >= 8804 && e <= 8807 || e === 8810 || e === 8811 || e === 8814 || e === 8815 || e === 8834 || e === 8835 || e === 8838 || e === 8839 || e === 8853 || e === 8857 || e === 8869 || e === 8895 || e === 8978 || e >= 9312 && e <= 9449 || e >= 9451 && e <= 9547 || e >= 9552 && e <= 9587 || e >= 9600 && e <= 9615 || e >= 9618 && e <= 9621 || e === 9632 || e === 9633 || e >= 9635 && e <= 9641 || e === 9650 || e === 9651 || e === 9654 || e === 9655 || e === 9660 || e === 9661 || e === 9664 || e === 9665 || e >= 9670 && e <= 9672 || e === 9675 || e >= 9678 && e <= 9681 || e >= 9698 && e <= 9701 || e === 9711 || e === 9733 || e === 9734 || e === 9737 || e === 9742 || e === 9743 || e === 9756 || e === 9758 || e === 9792 || e === 9794 || e === 9824 || e === 9825 || e >= 9827 && e <= 9829 || e >= 9831 && e <= 9834 || e === 9836 || e === 9837 || e === 9839 || e === 9886 || e === 9887 || e === 9919 || e >= 9926 && e <= 9933 || e >= 9935 && e <= 9939 || e >= 9941 && e <= 9953 || e === 9955 || e === 9960 || e === 9961 || e >= 9963 && e <= 9969 || e === 9972 || e >= 9974 && e <= 9977 || e === 9979 || e === 9980 || e === 9982 || e === 9983 || e === 10045 || e >= 10102 && e <= 10111 || e >= 11094 && e <= 11097 || e >= 12872 && e <= 12879 || e >= 57344 && e <= 63743 || e >= 65024 && e <= 65039 || e === 65533 || e >= 127232 && e <= 127242 || e >= 127248 && e <= 127277 || e >= 127280 && e <= 127337 || e >= 127344 && e <= 127373 || e === 127375 || e === 127376 || e >= 127387 && e <= 127404 || e >= 917760 && e <= 917999 || e >= 983040 && e <= 1048573 || e >= 1048576 && e <= 1114109, pt = (e) => e === 12288 || e >= 65281 && e <= 65376 || e >= 65504 && e <= 65510, gt = (e) => e >= 4352 && e <= 4447 || e === 8986 || e === 8987 || e === 9001 || e === 9002 || e >= 9193 && e <= 9196 || e === 9200 || e === 9203 || e === 9725 || e === 9726 || e === 9748 || e === 9749 || e >= 9800 && e <= 9811 || e === 9855 || e === 9875 || e === 9889 || e === 9898 || e === 9899 || e === 9917 || e === 9918 || e === 9924 || e === 9925 || e === 9934 || e === 9940 || e === 9962 || e === 9970 || e === 9971 || e === 9973 || e === 9978 || e === 9981 || e === 9989 || e === 9994 || e === 9995 || e === 10024 || e === 10060 || e === 10062 || e >= 10067 && e <= 10069 || e === 10071 || e >= 10133 && e <= 10135 || e === 10160 || e === 10175 || e === 11035 || e === 11036 || e === 11088 || e === 11093 || e >= 11904 && e <= 11929 || e >= 11931 && e <= 12019 || e >= 12032 && e <= 12245 || e >= 12272 && e <= 12287 || e >= 12289 && e <= 12350 || e >= 12353 && e <= 12438 || e >= 12441 && e <= 12543 || e >= 12549 && e <= 12591 || e >= 12593 && e <= 12686 || e >= 12688 && e <= 12771 || e >= 12783 && e <= 12830 || e >= 12832 && e <= 12871 || e >= 12880 && e <= 19903 || e >= 19968 && e <= 42124 || e >= 42128 && e <= 42182 || e >= 43360 && e <= 43388 || e >= 44032 && e <= 55203 || e >= 63744 && e <= 64255 || e >= 65040 && e <= 65049 || e >= 65072 && e <= 65106 || e >= 65108 && e <= 65126 || e >= 65128 && e <= 65131 || e >= 94176 && e <= 94180 || e === 94192 || e === 94193 || e >= 94208 && e <= 100343 || e >= 100352 && e <= 101589 || e >= 101632 && e <= 101640 || e >= 110576 && e <= 110579 || e >= 110581 && e <= 110587 || e === 110589 || e === 110590 || e >= 110592 && e <= 110882 || e === 110898 || e >= 110928 && e <= 110930 || e === 110933 || e >= 110948 && e <= 110951 || e >= 110960 && e <= 111355 || e === 126980 || e === 127183 || e === 127374 || e >= 127377 && e <= 127386 || e >= 127488 && e <= 127490 || e >= 127504 && e <= 127547 || e >= 127552 && e <= 127560 || e === 127568 || e === 127569 || e >= 127584 && e <= 127589 || e >= 127744 && e <= 127776 || e >= 127789 && e <= 127797 || e >= 127799 && e <= 127868 || e >= 127870 && e <= 127891 || e >= 127904 && e <= 127946 || e >= 127951 && e <= 127955 || e >= 127968 && e <= 127984 || e === 127988 || e >= 127992 && e <= 128062 || e === 128064 || e >= 128066 && e <= 128252 || e >= 128255 && e <= 128317 || e >= 128331 && e <= 128334 || e >= 128336 && e <= 128359 || e === 128378 || e === 128405 || e === 128406 || e === 128420 || e >= 128507 && e <= 128591 || e >= 128640 && e <= 128709 || e === 128716 || e >= 128720 && e <= 128722 || e >= 128725 && e <= 128727 || e >= 128732 && e <= 128735 || e === 128747 || e === 128748 || e >= 128756 && e <= 128764 || e >= 128992 && e <= 129003 || e === 129008 || e >= 129292 && e <= 129338 || e >= 129340 && e <= 129349 || e >= 129351 && e <= 129535 || e >= 129648 && e <= 129660 || e >= 129664 && e <= 129672 || e >= 129680 && e <= 129725 || e >= 129727 && e <= 129733 || e >= 129742 && e <= 129755 || e >= 129760 && e <= 129768 || e >= 129776 && e <= 129784 || e >= 131072 && e <= 196605 || e >= 196608 && e <= 262141, ve = /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/y, re = /[\x00-\x08\x0A-\x1F\x7F-\x9F]{1,1000}/y, ie = /\t{1,1000}/y, Ae = /[\u{1F1E6}-\u{1F1FF}]{2}|\u{1F3F4}[\u{E0061}-\u{E007A}]{2}[\u{E0030}-\u{E0039}\u{E0061}-\u{E007A}]{1,3}\u{E007F}|(?:\p{Emoji}\uFE0F\u20E3?|\p{Emoji_Modifier_Base}\p{Emoji_Modifier}?|\p{Emoji_Presentation})(?:\u200D(?:\p{Emoji_Modifier_Base}\p{Emoji_Modifier}?|\p{Emoji_Presentation}|\p{Emoji}\uFE0F\u20E3?))*/uy, ne = /(?:[\x20-\x7E\xA0-\xFF](?!\uFE0F)){1,1000}/y, ft = /\p{M}+/gu, Ft = {
	limit: Infinity,
	ellipsis: ""
}, Le = (e, r = {}, s = {}) => {
	const i = r.limit ?? Infinity, n = r.ellipsis ?? "", o = r?.ellipsisWidth ?? (n ? Le(n, Ft, s).width : 0), u = s.ansiWidth ?? 0, l = s.controlWidth ?? 0, a = s.tabWidth ?? 8, d = s.ambiguousWidth ?? 1, g = s.emojiWidth ?? 2, E = s.fullWidthWidth ?? 2, p = s.regularWidth ?? 1, y = s.wideWidth ?? 2;
	let $ = 0, c = 0, m = e.length, f = 0, F = !1, v = m, S = Math.max(0, i - o), B = 0, b = 0, A = 0, C = 0;
	e: for (;;) {
		if (b > B || c >= m && c > $) {
			const _ = e.slice(B, b) || e.slice($, c);
			f = 0;
			for (const D of _.replaceAll(ft, "")) {
				const T = D.codePointAt(0) || 0;
				if (pt(T) ? C = E : gt(T) ? C = y : d !== p && mt(T) ? C = d : C = p, A + C > S && (v = Math.min(v, Math.max(B, $) + f)), A + C > i) {
					F = !0;
					break e;
				}
				f += D.length, A += C;
			}
			B = b = 0;
		}
		if (c >= m) break;
		if (ne.lastIndex = c, ne.test(e)) {
			if (f = ne.lastIndex - c, C = f * p, A + C > S && (v = Math.min(v, c + Math.floor((S - A) / p))), A + C > i) {
				F = !0;
				break;
			}
			A += C, B = $, b = c, c = $ = ne.lastIndex;
			continue;
		}
		if (ve.lastIndex = c, ve.test(e)) {
			if (A + u > S && (v = Math.min(v, c)), A + u > i) {
				F = !0;
				break;
			}
			A += u, B = $, b = c, c = $ = ve.lastIndex;
			continue;
		}
		if (re.lastIndex = c, re.test(e)) {
			if (f = re.lastIndex - c, C = f * l, A + C > S && (v = Math.min(v, c + Math.floor((S - A) / l))), A + C > i) {
				F = !0;
				break;
			}
			A += C, B = $, b = c, c = $ = re.lastIndex;
			continue;
		}
		if (ie.lastIndex = c, ie.test(e)) {
			if (f = ie.lastIndex - c, C = f * a, A + C > S && (v = Math.min(v, c + Math.floor((S - A) / a))), A + C > i) {
				F = !0;
				break;
			}
			A += C, B = $, b = c, c = $ = ie.lastIndex;
			continue;
		}
		if (Ae.lastIndex = c, Ae.test(e)) {
			if (A + g > S && (v = Math.min(v, c)), A + g > i) {
				F = !0;
				break;
			}
			A += g, B = $, b = c, c = $ = Ae.lastIndex;
			continue;
		}
		c += 1;
	}
	return {
		width: F ? S : A,
		index: F ? v : m,
		truncated: F,
		ellipsed: F && i >= o
	};
}, yt = {
	limit: Infinity,
	ellipsis: "",
	ellipsisWidth: 0
}, M = (e, r = {}) => Le(e, yt, r).width, ae = "\x1B", je = "", Et = 39, Ce = "\x07", Ve = "[", vt = "]", ke = "m", we = `${vt}8;;`, Ge = new RegExp(`(?:\\${Ve}(?<code>\\d+)m|\\${we}(?<uri>.*)${Ce})`, "y"), At = (e) => {
	if (e >= 30 && e <= 37 || e >= 90 && e <= 97) return 39;
	if (e >= 40 && e <= 47 || e >= 100 && e <= 107) return 49;
	if (e === 1 || e === 2) return 22;
	if (e === 3) return 23;
	if (e === 4) return 24;
	if (e === 7) return 27;
	if (e === 8) return 28;
	if (e === 9) return 29;
	if (e === 0) return 0;
}, He = (e) => `${ae}${Ve}${e}${ke}`, Ue = (e) => `${ae}${we}${e}${Ce}`, Ct = (e) => e.map((r) => M(r)), Se = (e, r, s) => {
	const i = r[Symbol.iterator]();
	let n = !1, o = !1, u = e.at(-1), l = u === void 0 ? 0 : M(u), a = i.next(), d = i.next(), g = 0;
	for (; !a.done;) {
		const E = a.value, p = M(E);
		l + p <= s ? e[e.length - 1] += E : (e.push(E), l = 0), (E === ae || E === je) && (n = !0, o = r.startsWith(we, g + 1)), n ? o ? E === Ce && (n = !1, o = !1) : E === ke && (n = !1) : (l += p, l === s && !d.done && (e.push(""), l = 0)), a = d, d = i.next(), g += E.length;
	}
	u = e.at(-1), !l && u !== void 0 && u.length > 0 && e.length > 1 && (e[e.length - 2] += e.pop());
}, wt = (e) => {
	const r = e.split(" ");
	let s = r.length;
	for (; s > 0 && !(M(r[s - 1]) > 0);) s--;
	return s === r.length ? e : r.slice(0, s).join(" ") + r.slice(s).join("");
}, St = (e, r, s = {}) => {
	if (s.trim !== !1 && e.trim() === "") return "";
	let i = "", n, o;
	const u = e.split(" "), l = Ct(u);
	let a = [""];
	for (const [$, c] of u.entries()) {
		s.trim !== !1 && (a[a.length - 1] = (a.at(-1) ?? "").trimStart());
		let m = M(a.at(-1) ?? "");
		if ($ !== 0 && (m >= r && (s.wordWrap === !1 || s.trim === !1) && (a.push(""), m = 0), (m > 0 || s.trim === !1) && (a[a.length - 1] += " ", m++)), s.hard && l[$] > r) {
			const f = r - m, F = 1 + Math.floor((l[$] - f - 1) / r);
			Math.floor((l[$] - 1) / r) < F && a.push(""), Se(a, c, r);
			continue;
		}
		if (m + l[$] > r && m > 0 && l[$] > 0) {
			if (s.wordWrap === !1 && m < r) {
				Se(a, c, r);
				continue;
			}
			a.push("");
		}
		if (m + l[$] > r && s.wordWrap === !1) {
			Se(a, c, r);
			continue;
		}
		a[a.length - 1] += c;
	}
	s.trim !== !1 && (a = a.map(($) => wt($)));
	const d = a.join(`
`), g = d[Symbol.iterator]();
	let E = g.next(), p = g.next(), y = 0;
	for (; !E.done;) {
		const $ = E.value, c = p.value;
		if (i += $, $ === ae || $ === je) {
			Ge.lastIndex = y + 1;
			const F = Ge.exec(d)?.groups;
			if (F?.code !== void 0) {
				const v = Number.parseFloat(F.code);
				n = v === Et ? void 0 : v;
			} else F?.uri !== void 0 && (o = F.uri.length === 0 ? void 0 : F.uri);
		}
		const m = n ? At(n) : void 0;
		c === `
` ? (o && (i += Ue("")), n && m && (i += He(m))) : $ === `
` && (n && m && (i += He(n)), o && (i += Ue(o))), y += $.length, E = p, p = g.next();
	}
	return i;
};
function q(e, r, s) {
	return String(e).normalize().replaceAll(`\r
`, `
`).split(`
`).map((i) => St(i, r, s)).join(`
`);
}
const It = (e, r, s, i, n) => {
	let o = r, u = 0;
	for (let l = s; l < i; l++) {
		const a = e[l];
		if (o = o - a.length, u++, o <= n) break;
	}
	return {
		lineCount: o,
		removals: u
	};
}, J = (e) => {
	const { cursor: r, options: s, style: i } = e, n = e.output ?? process.stdout, o = rt(n), u = e.columnPadding ?? 0, l = e.rowPadding ?? 4, a = o - u, d = nt(n), g = import_picocolors.default.dim("..."), E = e.maxItems ?? Number.POSITIVE_INFINITY, p = Math.max(d - l, 0), y = Math.max(Math.min(E, p), 5);
	let $ = 0;
	r >= y - 3 && ($ = Math.max(Math.min(r - y + 3, s.length - y), 0));
	let c = y < s.length && $ > 0, m = y < s.length && $ + y < s.length;
	const f = Math.min($ + y, s.length), F = [];
	let v = 0;
	c && v++, m && v++;
	const S = $ + (c ? 1 : 0), B = f - (m ? 1 : 0);
	for (let A = S; A < B; A++) {
		const C = q(i(s[A], A === r), a, {
			hard: !0,
			trim: !1
		}).split(`
`);
		F.push(C), v += C.length;
	}
	if (v > p) {
		let A = 0, C = 0, _ = v;
		const D = r - S, T = (W, I) => It(F, _, W, I, p);
		c ? ({lineCount: _, removals: A} = T(0, D), _ > p && ({lineCount: _, removals: C} = T(D + 1, F.length))) : ({lineCount: _, removals: C} = T(D + 1, F.length), _ > p && ({lineCount: _, removals: A} = T(0, D))), A > 0 && (c = !0, F.splice(0, A)), C > 0 && (m = !0, F.splice(F.length - C, C));
	}
	const b = [];
	c && b.push(g);
	for (const A of F) for (const C of A) b.push(C);
	return m && b.push(g), b;
};
function Ke(e) {
	return e.label ?? String(e.value ?? "");
}
function qe(e, r) {
	if (!e) return !0;
	const s = (r.label ?? String(r.value ?? "")).toLowerCase(), i = (r.hint ?? "").toLowerCase(), n = String(r.value).toLowerCase(), o = e.toLowerCase();
	return s.includes(o) || i.includes(o) || n.includes(o);
}
function Bt(e, r) {
	const s = [];
	for (const i of r) e.includes(i.value) && s.push(i);
	return s;
}
const Je = (e) => new Vt$1({
	options: e.options,
	initialValue: e.initialValue ? [e.initialValue] : void 0,
	initialUserInput: e.initialUserInput,
	filter: e.filter ?? ((r, s) => qe(r, s)),
	signal: e.signal,
	input: e.input,
	output: e.output,
	validate: e.validate,
	render() {
		const r = [`${import_picocolors.default.gray(h)}`, `${N(this.state)}  ${e.message}`], s = this.userInput, i = this.options, n = e.placeholder, o = s === "" && n !== void 0;
		switch (this.state) {
			case "submit": {
				const u = Bt(this.selectedValues, i), l = u.length > 0 ? `  ${import_picocolors.default.dim(u.map(Ke).join(", "))}` : "";
				return `${r.join(`
`)}
${import_picocolors.default.gray(h)}${l}`;
			}
			case "cancel": {
				const u = s ? `  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(s))}` : "";
				return `${r.join(`
`)}
${import_picocolors.default.gray(h)}${u}`;
			}
			default: {
				const u = `${(this.state === "error" ? import_picocolors.default.yellow : import_picocolors.default.cyan)(h)}  `, l = (this.state === "error" ? import_picocolors.default.yellow : import_picocolors.default.cyan)(x);
				let a = "";
				if (this.isNavigating || o) {
					const c = o ? n : s;
					a = c !== "" ? ` ${import_picocolors.default.dim(c)}` : "";
				} else a = ` ${this.userInputWithCursor}`;
				const d = this.filteredOptions.length !== i.length ? import_picocolors.default.dim(` (${this.filteredOptions.length} match${this.filteredOptions.length === 1 ? "" : "es"})`) : "", g = this.filteredOptions.length === 0 && s ? [`${u}${import_picocolors.default.yellow("No matches found")}`] : [], E = this.state === "error" ? [`${u}${import_picocolors.default.yellow(this.error)}`] : [];
				r.push(`${u.trimEnd()}`, `${u}${import_picocolors.default.dim("Search:")}${a}${d}`, ...g, ...E);
				const p = [
					`${import_picocolors.default.dim("↑/↓")} to select`,
					`${import_picocolors.default.dim("Enter:")} confirm`,
					`${import_picocolors.default.dim("Type:")} to search`
				], y = [`${u}${import_picocolors.default.dim(p.join(" • "))}`, `${l}`], $ = this.filteredOptions.length === 0 ? [] : J({
					cursor: this.cursor,
					options: this.filteredOptions,
					columnPadding: 3,
					rowPadding: r.length + y.length,
					style: (c, m) => {
						const f = Ke(c), F = c.hint && c.value === this.focusedValue ? import_picocolors.default.dim(` (${c.hint})`) : "";
						return m ? `${import_picocolors.default.green(Y)} ${f}${F}` : `${import_picocolors.default.dim(K)} ${import_picocolors.default.dim(f)}${F}`;
					},
					maxItems: e.maxItems,
					output: e.output
				});
				return [
					...r,
					...$.map((c) => `${u}${c}`),
					...y
				].join(`
`);
			}
		}
	}
}).prompt(), bt = (e) => {
	const r = (i, n, o, u) => {
		const l = o.includes(i.value), a = i.label ?? String(i.value ?? ""), d = i.hint && u !== void 0 && i.value === u ? import_picocolors.default.dim(` (${i.hint})`) : "", g = l ? import_picocolors.default.green(G) : import_picocolors.default.dim(z);
		return n ? `${g} ${a}${d}` : `${g} ${import_picocolors.default.dim(a)}`;
	}, s = new Vt$1({
		options: e.options,
		multiple: !0,
		filter: e.filter ?? ((i, n) => qe(i, n)),
		validate: () => {
			if (e.required && s.selectedValues.length === 0) return "Please select at least one item";
		},
		initialValue: e.initialValues,
		signal: e.signal,
		input: e.input,
		output: e.output,
		render() {
			const i = `${import_picocolors.default.gray(h)}
${N(this.state)}  ${e.message}
`, n = this.userInput, o = e.placeholder, u = n === "" && o !== void 0, l = this.isNavigating || u ? import_picocolors.default.dim(u ? o : n) : this.userInputWithCursor, a = this.options, d = this.filteredOptions.length !== a.length ? import_picocolors.default.dim(` (${this.filteredOptions.length} match${this.filteredOptions.length === 1 ? "" : "es"})`) : "";
			switch (this.state) {
				case "submit": return `${i}${import_picocolors.default.gray(h)}  ${import_picocolors.default.dim(`${this.selectedValues.length} items selected`)}`;
				case "cancel": return `${i}${import_picocolors.default.gray(h)}  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(n))}`;
				default: {
					const g = this.state === "error" ? import_picocolors.default.yellow : import_picocolors.default.cyan, E = [
						`${import_picocolors.default.dim("↑/↓")} to navigate`,
						`${import_picocolors.default.dim(this.isNavigating ? "Space/Tab:" : "Tab:")} select`,
						`${import_picocolors.default.dim("Enter:")} confirm`,
						`${import_picocolors.default.dim("Type:")} to search`
					], p = this.filteredOptions.length === 0 && n ? [`${g(h)}  ${import_picocolors.default.yellow("No matches found")}`] : [], y = this.state === "error" ? [`${g(h)}  ${import_picocolors.default.yellow(this.error)}`] : [], $ = [
						...`${i}${g(h)}`.split(`
`),
						`${g(h)}  ${import_picocolors.default.dim("Search:")} ${l}${d}`,
						...p,
						...y
					], c = [`${g(h)}  ${import_picocolors.default.dim(E.join(" • "))}`, `${g(x)}`], m = J({
						cursor: this.cursor,
						options: this.filteredOptions,
						style: (f, F) => r(f, F, this.selectedValues, this.focusedValue),
						maxItems: e.maxItems,
						output: e.output,
						rowPadding: $.length + c.length
					});
					return [
						...$,
						...m.map((f) => `${g(h)}  ${f}`),
						...c
					].join(`
`);
				}
			}
		}
	});
	return s.prompt();
}, xt = [
	We,
	he,
	pe,
	me
], _t = [
	$e,
	Re,
	x,
	Oe
];
function Xe(e, r, s, i) {
	let n = s, o = s;
	return i === "center" ? n = Math.floor((r - e) / 2) : i === "right" && (n = r - e - s), o = r - n - e, [n, o];
}
const Dt = (e) => e, Tt = (e = "", r = "", s) => {
	const i = s?.output ?? process.stdout, n = rt(i), o = 2, u = s?.titlePadding ?? 1, l = s?.contentPadding ?? 2, a = s?.width === void 0 || s.width === "auto" ? 1 : Math.min(1, s.width), d = (s?.withGuide ?? _.withGuide) !== !1 ? `${h} ` : "", g = s?.formatBorder ?? Dt, E = (s?.rounded ? xt : _t).map(g), p = g(se), y = g(h), $ = M(d), c = M(r), m = n - $;
	let f = Math.floor(n * a) - $;
	if (s?.width === "auto") {
		const _ = e.split(`
`);
		let D = c + u * 2;
		for (const W of _) {
			const I = M(W) + l * 2;
			I > D && (D = I);
		}
		const T = D + o;
		T < f && (f = T);
	}
	f % 2 !== 0 && (f < m ? f++ : f--);
	const F = f - o, v = F - u * 2, S = c > v ? `${r.slice(0, v - 3)}...` : r, [B, b] = Xe(M(S), F, u, s?.titleAlign), A = q(e, F - l * 2, {
		hard: !0,
		trim: !1
	});
	i.write(`${d}${E[0]}${p.repeat(B)}${S}${p.repeat(b)}${E[1]}
`);
	const C = A.split(`
`);
	for (const _ of C) {
		const [D, T] = Xe(M(_), F, l, s?.contentAlign);
		i.write(`${d}${y}${" ".repeat(D)}${_}${" ".repeat(T)}${y}
`);
	}
	i.write(`${d}${E[2]}${p.repeat(F)}${E[3]}
`);
}, Mt = (e) => {
	const r = e.active ?? "Yes", s = e.inactive ?? "No";
	return new kt$1({
		active: r,
		inactive: s,
		signal: e.signal,
		input: e.input,
		output: e.output,
		initialValue: e.initialValue ?? !0,
		render() {
			const i = `${import_picocolors.default.gray(h)}
${N(this.state)}  ${e.message}
`, n = this.value ? r : s;
			switch (this.state) {
				case "submit": return `${i}${import_picocolors.default.gray(h)}  ${import_picocolors.default.dim(n)}`;
				case "cancel": return `${i}${import_picocolors.default.gray(h)}  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(n))}
${import_picocolors.default.gray(h)}`;
				default: return `${i}${import_picocolors.default.cyan(h)}  ${this.value ? `${import_picocolors.default.green(Y)} ${r}` : `${import_picocolors.default.dim(K)} ${import_picocolors.default.dim(r)}`} ${import_picocolors.default.dim("/")} ${this.value ? `${import_picocolors.default.dim(K)} ${import_picocolors.default.dim(s)}` : `${import_picocolors.default.green(Y)} ${s}`}
${import_picocolors.default.cyan(x)}
`;
			}
		}
	}).prompt();
}, Rt = async (e, r) => {
	const s = {}, i = Object.keys(e);
	for (const n of i) {
		const o = e[n], u = await o({ results: s })?.catch((l) => {
			throw l;
		});
		if (typeof r?.onCancel == "function" && Ct$1(u)) {
			s[n] = "canceled", r.onCancel({ results: s });
			continue;
		}
		s[n] = u;
	}
	return s;
}, Ot = (e) => {
	const { selectableGroups: r = !0, groupSpacing: s = 0 } = e, i = (o, u, l = []) => {
		const a = o.label ?? String(o.value), d = typeof o.group == "string", g = d && (l[l.indexOf(o) + 1] ?? { group: !0 }), E = d && g && g.group === !0, p = d ? r ? `${E ? x : h} ` : "  " : "";
		let y = "";
		if (s > 0 && !d) {
			const c = `
${import_picocolors.default.cyan(h)}`;
			y = `${c.repeat(s - 1)}${c}  `;
		}
		if (u === "active") return `${y}${import_picocolors.default.dim(p)}${import_picocolors.default.cyan(te)} ${a}${o.hint ? ` ${import_picocolors.default.dim(`(${o.hint})`)}` : ""}`;
		if (u === "group-active") return `${y}${p}${import_picocolors.default.cyan(te)} ${import_picocolors.default.dim(a)}`;
		if (u === "group-active-selected") return `${y}${p}${import_picocolors.default.green(G)} ${import_picocolors.default.dim(a)}`;
		if (u === "selected") {
			const c = d || r ? import_picocolors.default.green(G) : "";
			return `${y}${import_picocolors.default.dim(p)}${c} ${import_picocolors.default.dim(a)}${o.hint ? ` ${import_picocolors.default.dim(`(${o.hint})`)}` : ""}`;
		}
		if (u === "cancelled") return `${import_picocolors.default.strikethrough(import_picocolors.default.dim(a))}`;
		if (u === "active-selected") return `${y}${import_picocolors.default.dim(p)}${import_picocolors.default.green(G)} ${a}${o.hint ? ` ${import_picocolors.default.dim(`(${o.hint})`)}` : ""}`;
		if (u === "submitted") return `${import_picocolors.default.dim(a)}`;
		const $ = d || r ? import_picocolors.default.dim(z) : "";
		return `${y}${import_picocolors.default.dim(p)}${$} ${import_picocolors.default.dim(a)}`;
	}, n = e.required ?? !0;
	return new yt$1({
		options: e.options,
		signal: e.signal,
		input: e.input,
		output: e.output,
		initialValues: e.initialValues,
		required: n,
		cursorAt: e.cursorAt,
		selectableGroups: r,
		validate(o) {
			if (n && (o === void 0 || o.length === 0)) return `Please select at least one option.
${import_picocolors.default.reset(import_picocolors.default.dim(`Press ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" space ")))} to select, ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" enter ")))} to submit`))}`;
		},
		render() {
			const o = `${import_picocolors.default.gray(h)}
${N(this.state)}  ${e.message}
`, u = this.value ?? [];
			switch (this.state) {
				case "submit": {
					const l = this.options.filter(({ value: d }) => u.includes(d)).map((d) => i(d, "submitted")), a = l.length === 0 ? "" : `  ${l.join(import_picocolors.default.dim(", "))}`;
					return `${o}${import_picocolors.default.gray(h)}${a}`;
				}
				case "cancel": {
					const l = this.options.filter(({ value: a }) => u.includes(a)).map((a) => i(a, "cancelled")).join(import_picocolors.default.dim(", "));
					return `${o}${import_picocolors.default.gray(h)}  ${l.trim() ? `${l}
${import_picocolors.default.gray(h)}` : ""}`;
				}
				case "error": {
					const l = this.error.split(`
`).map((a, d) => d === 0 ? `${import_picocolors.default.yellow(x)}  ${import_picocolors.default.yellow(a)}` : `   ${a}`).join(`
`);
					return `${o}${import_picocolors.default.yellow(h)}  ${this.options.map((a, d, g) => {
						const E = u.includes(a.value) || a.group === !0 && this.isGroupSelected(`${a.value}`), p = d === this.cursor;
						return !p && typeof a.group == "string" && this.options[this.cursor].value === a.group ? i(a, E ? "group-active-selected" : "group-active", g) : p && E ? i(a, "active-selected", g) : E ? i(a, "selected", g) : i(a, p ? "active" : "inactive", g);
					}).join(`
${import_picocolors.default.yellow(h)}  `)}
${l}
`;
				}
				default: {
					const l = this.options.map((d, g, E) => {
						const p = u.includes(d.value) || d.group === !0 && this.isGroupSelected(`${d.value}`), y = g === this.cursor, $ = !y && typeof d.group == "string" && this.options[this.cursor].value === d.group;
						let c = "";
						return $ ? c = i(d, p ? "group-active-selected" : "group-active", E) : y && p ? c = i(d, "active-selected", E) : p ? c = i(d, "selected", E) : c = i(d, y ? "active" : "inactive", E), `${g !== 0 && !c.startsWith(`
`) ? "  " : ""}${c}`;
					}).join(`
${import_picocolors.default.cyan(h)}`), a = l.startsWith(`
`) ? "" : "  ";
					return `${o}${import_picocolors.default.cyan(h)}${a}${l}
${import_picocolors.default.cyan(x)}
`;
				}
			}
		}
	}).prompt();
}, R = {
	message: (e = [], { symbol: r = import_picocolors.default.gray(h), secondarySymbol: s = import_picocolors.default.gray(h), output: i = process.stdout, spacing: n = 1, withGuide: o } = {}) => {
		const u = [], l = (o ?? _.withGuide) !== !1, a = l ? s : "", d = l ? `${r}  ` : "", g = l ? `${s}  ` : "";
		for (let p = 0; p < n; p++) u.push(a);
		const E = Array.isArray(e) ? e : e.split(`
`);
		if (E.length > 0) {
			const [p, ...y] = E;
			p.length > 0 ? u.push(`${d}${p}`) : u.push(l ? r : "");
			for (const $ of y) $.length > 0 ? u.push(`${g}${$}`) : u.push(l ? s : "");
		}
		i.write(`${u.join(`
`)}
`);
	},
	info: (e, r) => {
		R.message(e, {
			...r,
			symbol: import_picocolors.default.blue(ge)
		});
	},
	success: (e, r) => {
		R.message(e, {
			...r,
			symbol: import_picocolors.default.green(fe)
		});
	},
	step: (e, r) => {
		R.message(e, {
			...r,
			symbol: import_picocolors.default.green(k)
		});
	},
	warn: (e, r) => {
		R.message(e, {
			...r,
			symbol: import_picocolors.default.yellow(Fe)
		});
	},
	warning: (e, r) => {
		R.warn(e, r);
	},
	error: (e, r) => {
		R.message(e, {
			...r,
			symbol: import_picocolors.default.red(ye)
		});
	}
}, Pt = (e = "", r) => {
	(r?.output ?? process.stdout).write(`${import_picocolors.default.gray(x)}  ${import_picocolors.default.red(e)}

`);
}, Nt = (e = "", r) => {
	(r?.output ?? process.stdout).write(`${import_picocolors.default.gray($e)}  ${e}
`);
}, Wt = (e = "", r) => {
	(r?.output ?? process.stdout).write(`${import_picocolors.default.gray(h)}
${import_picocolors.default.gray(x)}  ${e}

`);
}, Q = (e, r) => e.split(`
`).map((s) => r(s)).join(`
`), Lt = (e) => {
	const r = (i, n) => {
		const o = i.label ?? String(i.value);
		return n === "disabled" ? `${import_picocolors.default.gray(z)} ${Q(o, (u) => import_picocolors.default.strikethrough(import_picocolors.default.gray(u)))}${i.hint ? ` ${import_picocolors.default.dim(`(${i.hint ?? "disabled"})`)}` : ""}` : n === "active" ? `${import_picocolors.default.cyan(te)} ${o}${i.hint ? ` ${import_picocolors.default.dim(`(${i.hint})`)}` : ""}` : n === "selected" ? `${import_picocolors.default.green(G)} ${Q(o, import_picocolors.default.dim)}${i.hint ? ` ${import_picocolors.default.dim(`(${i.hint})`)}` : ""}` : n === "cancelled" ? `${Q(o, (u) => import_picocolors.default.strikethrough(import_picocolors.default.dim(u)))}` : n === "active-selected" ? `${import_picocolors.default.green(G)} ${o}${i.hint ? ` ${import_picocolors.default.dim(`(${i.hint})`)}` : ""}` : n === "submitted" ? `${Q(o, import_picocolors.default.dim)}` : `${import_picocolors.default.dim(z)} ${Q(o, import_picocolors.default.dim)}`;
	}, s = e.required ?? !0;
	return new Mt$1({
		options: e.options,
		signal: e.signal,
		input: e.input,
		output: e.output,
		initialValues: e.initialValues,
		required: s,
		cursorAt: e.cursorAt,
		validate(i) {
			if (s && (i === void 0 || i.length === 0)) return `Please select at least one option.
${import_picocolors.default.reset(import_picocolors.default.dim(`Press ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" space ")))} to select, ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" enter ")))} to submit`))}`;
		},
		render() {
			const i = Bt$1(e.output, e.message, `${Ee(this.state)}  `, `${N(this.state)}  `), n = `${import_picocolors.default.gray(h)}
${i}
`, o = this.value ?? [], u = (l, a) => {
				if (l.disabled) return r(l, "disabled");
				const d = o.includes(l.value);
				return a && d ? r(l, "active-selected") : d ? r(l, "selected") : r(l, a ? "active" : "inactive");
			};
			switch (this.state) {
				case "submit": {
					const l = this.options.filter(({ value: d }) => o.includes(d)).map((d) => r(d, "submitted")).join(import_picocolors.default.dim(", ")) || import_picocolors.default.dim("none");
					return `${n}${Bt$1(e.output, l, `${import_picocolors.default.gray(h)}  `)}`;
				}
				case "cancel": {
					const l = this.options.filter(({ value: d }) => o.includes(d)).map((d) => r(d, "cancelled")).join(import_picocolors.default.dim(", "));
					if (l.trim() === "") return `${n}${import_picocolors.default.gray(h)}`;
					return `${n}${Bt$1(e.output, l, `${import_picocolors.default.gray(h)}  `)}
${import_picocolors.default.gray(h)}`;
				}
				case "error": {
					const l = `${import_picocolors.default.yellow(h)}  `, a = this.error.split(`
`).map((E, p) => p === 0 ? `${import_picocolors.default.yellow(x)}  ${import_picocolors.default.yellow(E)}` : `   ${E}`).join(`
`), d = n.split(`
`).length, g = a.split(`
`).length + 1;
					return `${n}${l}${J({
						output: e.output,
						options: this.options,
						cursor: this.cursor,
						maxItems: e.maxItems,
						columnPadding: l.length,
						rowPadding: d + g,
						style: u
					}).join(`
${l}`)}
${a}
`;
				}
				default: {
					const l = `${import_picocolors.default.cyan(h)}  `, a = n.split(`
`).length;
					return `${n}${l}${J({
						output: e.output,
						options: this.options,
						cursor: this.cursor,
						maxItems: e.maxItems,
						columnPadding: l.length,
						rowPadding: a + 2,
						style: u
					}).join(`
${l}`)}
${import_picocolors.default.cyan(x)}
`;
				}
			}
		}
	}).prompt();
}, jt = (e) => import_picocolors.default.dim(e), Vt = (e, r, s) => {
	const i = {
		hard: !0,
		trim: !1
	}, n = q(e, r, i).split(`
`), o = n.reduce((a, d) => Math.max(M(d), a), 0);
	return q(e, r - (n.map(s).reduce((a, d) => Math.max(M(d), a), 0) - o), i);
}, kt = (e = "", r = "", s) => {
	const i = s?.output ?? process$1.stdout, n = (s?.withGuide ?? _.withGuide) !== !1, o = s?.format ?? jt, u = [
		"",
		...Vt(e, rt(i) - 6, o).split(`
`).map(o),
		""
	], l = M(r), a = Math.max(u.reduce((p, y) => {
		const $ = M(y);
		return $ > p ? $ : p;
	}, 0), l) + 2, d = u.map((p) => `${import_picocolors.default.gray(h)}  ${p}${" ".repeat(a - M(p))}${import_picocolors.default.gray(h)}`).join(`
`), g = n ? `${import_picocolors.default.gray(h)}
` : "", E = n ? Ne : pe;
	i.write(`${g}${import_picocolors.default.green(k)}  ${import_picocolors.default.reset(r)} ${import_picocolors.default.gray(se.repeat(Math.max(a - l - 1, 1)) + he)}
${d}
${import_picocolors.default.gray(E + se.repeat(a + 2) + me)}
`);
}, Gt = (e) => new Lt$1({
	validate: e.validate,
	mask: e.mask ?? Pe,
	signal: e.signal,
	input: e.input,
	output: e.output,
	render() {
		const r = `${import_picocolors.default.gray(h)}
${N(this.state)}  ${e.message}
`, s = this.userInputWithCursor, i = this.masked;
		switch (this.state) {
			case "error": {
				const n = i ? `  ${i}` : "";
				return e.clearOnError && this.clear(), `${r.trim()}
${import_picocolors.default.yellow(h)}${n}
${import_picocolors.default.yellow(x)}  ${import_picocolors.default.yellow(this.error)}
`;
			}
			case "submit": {
				const n = i ? `  ${import_picocolors.default.dim(i)}` : "";
				return `${r}${import_picocolors.default.gray(h)}${n}`;
			}
			case "cancel": {
				const n = i ? `  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(i))}` : "";
				return `${r}${import_picocolors.default.gray(h)}${n}${i ? `
${import_picocolors.default.gray(h)}` : ""}`;
			}
			default: return `${r}${import_picocolors.default.cyan(h)}  ${s}
${import_picocolors.default.cyan(x)}
`;
		}
	}
}).prompt(), Ht = (e) => {
	const r = e.validate;
	return Je({
		...e,
		initialUserInput: e.initialValue ?? e.root ?? process.cwd(),
		maxItems: 5,
		validate(s) {
			if (!Array.isArray(s)) {
				if (!s) return "Please select a path";
				if (r) return r(s);
			}
		},
		options() {
			const s = this.userInput;
			if (s === "") return [];
			try {
				let i;
				return existsSync(s) ? lstatSync(s).isDirectory() ? i = s : i = dirname(s) : i = dirname(s), readdirSync(i).map((n) => {
					const o = join(i, n);
					return {
						name: n,
						path: o,
						isDirectory: lstatSync(o).isDirectory()
					};
				}).filter(({ path: n, isDirectory: o }) => n.startsWith(s) && (e.directory || !o)).map((n) => ({ value: n.path }));
			} catch {
				return [];
			}
		}
	});
}, Ut = import_picocolors.default.magenta, Ie = ({ indicator: e = "dots", onCancel: r, output: s = process.stdout, cancelMessage: i, errorMessage: n, frames: o = ee ? [
	"◒",
	"◐",
	"◓",
	"◑"
] : [
	"•",
	"o",
	"O",
	"0"
], delay: u = ee ? 80 : 120, signal: l, ...a } = {}) => {
	const d = ue();
	let g, E, p = !1, y = !1, $ = "", c, m = performance.now();
	const f = rt(s), F = a?.styleFrame ?? Ut, v = (I) => {
		const O = I > 1 ? n ?? _.messages.error : i ?? _.messages.cancel;
		y = I === 1, p && (W(O, I), y && typeof r == "function" && r());
	}, S = () => v(2), B = () => v(1), b = () => {
		process.on("uncaughtExceptionMonitor", S), process.on("unhandledRejection", S), process.on("SIGINT", B), process.on("SIGTERM", B), process.on("exit", v), l && l.addEventListener("abort", B);
	}, A = () => {
		process.removeListener("uncaughtExceptionMonitor", S), process.removeListener("unhandledRejection", S), process.removeListener("SIGINT", B), process.removeListener("SIGTERM", B), process.removeListener("exit", v), l && l.removeEventListener("abort", B);
	}, C = () => {
		if (c === void 0) return;
		d && s.write(`
`);
		const I = q(c, f, {
			hard: !0,
			trim: !1
		}).split(`
`);
		I.length > 1 && s.write(import_src$1.cursor.up(I.length - 1)), s.write(import_src$1.cursor.to(0)), s.write(import_src$1.erase.down());
	}, _$1 = (I) => I.replace(/\.+$/, ""), D = (I) => {
		const O = (performance.now() - I) / 1e3, L = Math.floor(O / 60), j = Math.floor(O % 60);
		return L > 0 ? `[${L}m ${j}s]` : `[${j}s]`;
	}, T = (I = "") => {
		p = !0, g = xt$1({ output: s }), $ = _$1(I), m = performance.now(), s.write(`${import_picocolors.default.gray(h)}
`);
		let O = 0, L = 0;
		b(), E = setInterval(() => {
			if (d && $ === c) return;
			C(), c = $;
			const j = F(o[O]);
			let Z;
			if (d) Z = `${j}  ${$}...`;
			else if (e === "timer") Z = `${j}  ${$} ${D(m)}`;
			else {
				const Ze = ".".repeat(Math.floor(L)).slice(0, 3);
				Z = `${j}  ${$}${Ze}`;
			}
			const Qe = q(Z, f, {
				hard: !0,
				trim: !1
			});
			s.write(Qe), O = O + 1 < o.length ? O + 1 : 0, L = L < 4 ? L + .125 : 0;
		}, u);
	}, W = (I = "", O = 0, L = !1) => {
		if (!p) return;
		p = !1, clearInterval(E), C();
		const j = O === 0 ? import_picocolors.default.green(k) : O === 1 ? import_picocolors.default.red(ce) : import_picocolors.default.red(de);
		$ = I ?? $, L || (e === "timer" ? s.write(`${j}  ${$} ${D(m)}
`) : s.write(`${j}  ${$}
`)), A(), g();
	};
	return {
		start: T,
		stop: (I = "") => W(I, 0),
		message: (I = "") => {
			$ = _$1(I ?? $);
		},
		cancel: (I = "") => W(I, 1),
		error: (I = "") => W(I, 2),
		clear: () => W("", 0, !0),
		get isCancelled() {
			return y;
		}
	};
}, Ye = {
	light: w("─", "-"),
	heavy: w("━", "="),
	block: w("█", "#")
};
const oe = (e, r) => e.includes(`
`) ? e.split(`
`).map((s) => r(s)).join(`
`) : r(e), qt = (e) => {
	const r = (s, i) => {
		const n = s.label ?? String(s.value);
		switch (i) {
			case "disabled": return `${import_picocolors.default.gray(K)} ${oe(n, import_picocolors.default.gray)}${s.hint ? ` ${import_picocolors.default.dim(`(${s.hint ?? "disabled"})`)}` : ""}`;
			case "selected": return `${oe(n, import_picocolors.default.dim)}`;
			case "active": return `${import_picocolors.default.green(Y)} ${n}${s.hint ? ` ${import_picocolors.default.dim(`(${s.hint})`)}` : ""}`;
			case "cancelled": return `${oe(n, (o) => import_picocolors.default.strikethrough(import_picocolors.default.dim(o)))}`;
			default: return `${import_picocolors.default.dim(K)} ${oe(n, import_picocolors.default.dim)}`;
		}
	};
	return new Wt$1({
		options: e.options,
		signal: e.signal,
		input: e.input,
		output: e.output,
		initialValue: e.initialValue,
		render() {
			const s = `${N(this.state)}  `, i = `${Ee(this.state)}  `, n = Bt$1(e.output, e.message, i, s), o = `${import_picocolors.default.gray(h)}
${n}
`;
			switch (this.state) {
				case "submit": {
					const u = `${import_picocolors.default.gray(h)}  `;
					return `${o}${Bt$1(e.output, r(this.options[this.cursor], "selected"), u)}`;
				}
				case "cancel": {
					const u = `${import_picocolors.default.gray(h)}  `;
					return `${o}${Bt$1(e.output, r(this.options[this.cursor], "cancelled"), u)}
${import_picocolors.default.gray(h)}`;
				}
				default: {
					const u = `${import_picocolors.default.cyan(h)}  `, l = o.split(`
`).length;
					return `${o}${u}${J({
						output: e.output,
						cursor: this.cursor,
						options: this.options,
						maxItems: e.maxItems,
						columnPadding: u.length,
						rowPadding: l + 2,
						style: (a, d) => r(a, a.disabled ? "disabled" : d ? "active" : "inactive")
					}).join(`
${u}`)}
${import_picocolors.default.cyan(x)}
`;
				}
			}
		}
	}).prompt();
}, Jt = (e) => {
	const r = (s, i = "inactive") => {
		const n = s.label ?? String(s.value);
		return i === "selected" ? `${import_picocolors.default.dim(n)}` : i === "cancelled" ? `${import_picocolors.default.strikethrough(import_picocolors.default.dim(n))}` : i === "active" ? `${import_picocolors.default.bgCyan(import_picocolors.default.gray(` ${s.value} `))} ${n}${s.hint ? ` ${import_picocolors.default.dim(`(${s.hint})`)}` : ""}` : `${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(` ${s.value} `)))} ${n}${s.hint ? ` ${import_picocolors.default.dim(`(${s.hint})`)}` : ""}`;
	};
	return new Tt$1({
		options: e.options,
		signal: e.signal,
		input: e.input,
		output: e.output,
		initialValue: e.initialValue,
		caseSensitive: e.caseSensitive,
		render() {
			const s = `${import_picocolors.default.gray(h)}
${N(this.state)}  ${e.message}
`;
			switch (this.state) {
				case "submit": {
					const i = `${import_picocolors.default.gray(h)}  `, n = this.options.find((u) => u.value === this.value) ?? e.options[0];
					return `${s}${Bt$1(e.output, r(n, "selected"), i)}`;
				}
				case "cancel": {
					const i = `${import_picocolors.default.gray(h)}  `;
					return `${s}${Bt$1(e.output, r(this.options[0], "cancelled"), i)}
${import_picocolors.default.gray(h)}`;
				}
				default: {
					const i = `${import_picocolors.default.cyan(h)}  `;
					return `${s}${this.options.map((o, u) => Bt$1(e.output, r(o, u === this.cursor ? "active" : "inactive"), i)).join(`
`)}
${import_picocolors.default.cyan(x)}
`;
				}
			}
		}
	}).prompt();
}, ze = `${import_picocolors.default.gray(h)}  `, H = {
	message: async (e, { symbol: r = import_picocolors.default.gray(h) } = {}) => {
		process.stdout.write(`${import_picocolors.default.gray(h)}
${r}  `);
		let s = 3;
		for await (let i of e) {
			i = i.replace(/\n/g, `
${ze}`), i.includes(`
`) && (s = 3 + stripVTControlCharacters(i.slice(i.lastIndexOf(`
`))).length);
			const n = stripVTControlCharacters(i).length;
			s + n < process.stdout.columns ? (s += n, process.stdout.write(i)) : (process.stdout.write(`
${ze}${i.trimStart()}`), s = 3 + stripVTControlCharacters(i.trimStart()).length);
		}
		process.stdout.write(`
`);
	},
	info: (e) => H.message(e, { symbol: import_picocolors.default.blue(ge) }),
	success: (e) => H.message(e, { symbol: import_picocolors.default.green(fe) }),
	step: (e) => H.message(e, { symbol: import_picocolors.default.green(k) }),
	warn: (e) => H.message(e, { symbol: import_picocolors.default.yellow(Fe) }),
	warning: (e) => H.warn(e),
	error: (e) => H.message(e, { symbol: import_picocolors.default.red(ye) })
}, Xt = async (e, r) => {
	for (const s of e) {
		if (s.enabled === !1) continue;
		const i = Ie(r);
		i.start(s.title);
		const n = await s.task(i.message);
		i.stop(n || s.title);
	}
}, Yt = (e) => e.replace(/\x1b\[(?:\d+;)*\d*[ABCDEFGHfJKSTsu]|\x1b\[(s|u)/g, ""), zt = (e) => {
	const r = e.output ?? process.stdout, s = rt(r), i = import_picocolors.default.gray(h), n = e.spacing ?? 1, o = 3, u = e.retainLog === !0, l = !ue() && Te(r);
	r.write(`${i}
`), r.write(`${import_picocolors.default.green(k)}  ${e.title}
`);
	for (let m = 0; m < n; m++) r.write(`${i}
`);
	const a = [{
		value: "",
		full: ""
	}];
	let d = !1;
	const g = (m) => {
		if (a.length === 0) return;
		let f = 0;
		m && (f += n + 2);
		for (const F of a) {
			const { value: v, result: S } = F;
			let B = S?.message ?? v;
			if (B.length === 0) continue;
			S === void 0 && F.header !== void 0 && F.header !== "" && (B += `
${F.header}`);
			const b = B.split(`
`).reduce((A, C) => C === "" ? A + 1 : A + Math.ceil((C.length + o) / s), 0);
			f += b;
		}
		f > 0 && (f += 1, r.write(import_src$1.erase.lines(f)));
	}, E = (m, f, F) => {
		const v = F ? `${m.full}
${m.value}` : m.value;
		m.header !== void 0 && m.header !== "" && R.message(m.header.split(`
`).map(import_picocolors.default.bold), {
			output: r,
			secondarySymbol: i,
			symbol: i,
			spacing: 0
		}), R.message(v.split(`
`).map(import_picocolors.default.dim), {
			output: r,
			secondarySymbol: i,
			symbol: i,
			spacing: f ?? n
		});
	}, p = () => {
		for (const m of a) {
			const { header: f, value: F, full: v } = m;
			(f === void 0 || f.length === 0) && F.length === 0 || E(m, void 0, u === !0 && v.length > 0);
		}
	}, y = (m, f, F) => {
		if (g(!1), (F?.raw !== !0 || !d) && m.value !== "" && (m.value += `
`), m.value += Yt(f), d = F?.raw === !0, e.limit !== void 0) {
			const v = m.value.split(`
`), S = v.length - e.limit;
			if (S > 0) {
				const B = v.splice(0, S);
				u && (m.full += (m.full === "" ? "" : `
`) + B.join(`
`));
			}
			m.value = v.join(`
`);
		}
		l && $();
	}, $ = () => {
		for (const m of a) m.result ? m.result.status === "error" ? R.error(m.result.message, {
			output: r,
			secondarySymbol: i,
			spacing: 0
		}) : R.success(m.result.message, {
			output: r,
			secondarySymbol: i,
			spacing: 0
		}) : m.value !== "" && E(m, 0);
	}, c = (m, f) => {
		g(!1), m.result = f, l && $();
	};
	return {
		message(m, f) {
			y(a[0], m, f);
		},
		group(m) {
			const f = {
				header: m,
				value: "",
				full: ""
			};
			return a.push(f), {
				message(F, v) {
					y(f, F, v);
				},
				error(F) {
					c(f, {
						status: "error",
						message: F
					});
				},
				success(F) {
					c(f, {
						status: "success",
						message: F
					});
				}
			};
		},
		error(m, f) {
			g(!0), R.error(m, {
				output: r,
				secondarySymbol: i,
				spacing: 1
			}), f?.showLog !== !1 && p(), a.splice(1, a.length - 1), a[0].value = "", a[0].full = "";
		},
		success(m, f) {
			g(!0), R.success(m, {
				output: r,
				secondarySymbol: i,
				spacing: 1
			}), f?.showLog === !0 && p(), a.splice(1, a.length - 1), a[0].value = "", a[0].full = "";
		}
	};
}, Qt = (e) => new $t({
	validate: e.validate,
	placeholder: e.placeholder,
	defaultValue: e.defaultValue,
	initialValue: e.initialValue,
	output: e.output,
	signal: e.signal,
	input: e.input,
	render() {
		const r = (e?.withGuide ?? _.withGuide) !== !1, s = `${`${r ? `${import_picocolors.default.gray(h)}
` : ""}${N(this.state)}  `}${e.message}
`, i = e.placeholder ? import_picocolors.default.inverse(e.placeholder[0]) + import_picocolors.default.dim(e.placeholder.slice(1)) : import_picocolors.default.inverse(import_picocolors.default.hidden("_")), n = this.userInput ? this.userInputWithCursor : i, o = this.value ?? "";
		switch (this.state) {
			case "error": {
				const u = this.error ? `  ${import_picocolors.default.yellow(this.error)}` : "", l = r ? `${import_picocolors.default.yellow(h)}  ` : "", a = r ? import_picocolors.default.yellow(x) : "";
				return `${s.trim()}
${l}${n}
${a}${u}
`;
			}
			case "submit": {
				const u = o ? `  ${import_picocolors.default.dim(o)}` : "";
				return `${s}${r ? import_picocolors.default.gray(h) : ""}${u}`;
			}
			case "cancel": {
				const u = o ? `  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(o))}` : "", l = r ? import_picocolors.default.gray(h) : "";
				return `${s}${l}${u}${o.trim() ? `
${l}` : ""}`;
			}
			default: return `${s}${r ? `${import_picocolors.default.cyan(h)}  ` : ""}${n}
${r ? import_picocolors.default.cyan(x) : ""}
`;
		}
	}
}).prompt();

//#endregion
//#region ../../node_modules/.pnpm/ms@2.1.3/node_modules/ms/index.js
var require_ms = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* Helpers.
	*/
	var s = 1e3;
	var m = s * 60;
	var h = m * 60;
	var d = h * 24;
	var w = d * 7;
	var y = d * 365.25;
	/**
	* Parse or format the given `val`.
	*
	* Options:
	*
	*  - `long` verbose formatting [false]
	*
	* @param {String|Number} val
	* @param {Object} [options]
	* @throws {Error} throw an error if val is not a non-empty string or a number
	* @return {String|Number}
	* @api public
	*/
	module.exports = function(val, options) {
		options = options || {};
		var type = typeof val;
		if (type === "string" && val.length > 0) return parse(val);
		else if (type === "number" && isFinite(val)) return options.long ? fmtLong(val) : fmtShort(val);
		throw new Error("val is not a non-empty string or a valid number. val=" + JSON.stringify(val));
	};
	/**
	* Parse the given `str` and return milliseconds.
	*
	* @param {String} str
	* @return {Number}
	* @api private
	*/
	function parse(str) {
		str = String(str);
		if (str.length > 100) return;
		var match = /^(-?(?:\d+)?\.?\d+) *(milliseconds?|msecs?|ms|seconds?|secs?|s|minutes?|mins?|m|hours?|hrs?|h|days?|d|weeks?|w|years?|yrs?|y)?$/i.exec(str);
		if (!match) return;
		var n = parseFloat(match[1]);
		switch ((match[2] || "ms").toLowerCase()) {
			case "years":
			case "year":
			case "yrs":
			case "yr":
			case "y": return n * y;
			case "weeks":
			case "week":
			case "w": return n * w;
			case "days":
			case "day":
			case "d": return n * d;
			case "hours":
			case "hour":
			case "hrs":
			case "hr":
			case "h": return n * h;
			case "minutes":
			case "minute":
			case "mins":
			case "min":
			case "m": return n * m;
			case "seconds":
			case "second":
			case "secs":
			case "sec":
			case "s": return n * s;
			case "milliseconds":
			case "millisecond":
			case "msecs":
			case "msec":
			case "ms": return n;
			default: return;
		}
	}
	/**
	* Short format for `ms`.
	*
	* @param {Number} ms
	* @return {String}
	* @api private
	*/
	function fmtShort(ms) {
		var msAbs = Math.abs(ms);
		if (msAbs >= d) return Math.round(ms / d) + "d";
		if (msAbs >= h) return Math.round(ms / h) + "h";
		if (msAbs >= m) return Math.round(ms / m) + "m";
		if (msAbs >= s) return Math.round(ms / s) + "s";
		return ms + "ms";
	}
	/**
	* Long format for `ms`.
	*
	* @param {Number} ms
	* @return {String}
	* @api private
	*/
	function fmtLong(ms) {
		var msAbs = Math.abs(ms);
		if (msAbs >= d) return plural(ms, msAbs, d, "day");
		if (msAbs >= h) return plural(ms, msAbs, h, "hour");
		if (msAbs >= m) return plural(ms, msAbs, m, "minute");
		if (msAbs >= s) return plural(ms, msAbs, s, "second");
		return ms + " ms";
	}
	/**
	* Pluralization helper.
	*/
	function plural(ms, msAbs, n, name) {
		var isPlural = msAbs >= n * 1.5;
		return Math.round(ms / n) + " " + name + (isPlural ? "s" : "");
	}
}));

//#endregion
//#region ../../node_modules/.pnpm/debug@4.4.3/node_modules/debug/src/common.js
var require_common = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* This is the common logic for both the Node.js and web browser
	* implementations of `debug()`.
	*/
	function setup(env) {
		createDebug.debug = createDebug;
		createDebug.default = createDebug;
		createDebug.coerce = coerce;
		createDebug.disable = disable;
		createDebug.enable = enable;
		createDebug.enabled = enabled;
		createDebug.humanize = require_ms();
		createDebug.destroy = destroy;
		Object.keys(env).forEach((key) => {
			createDebug[key] = env[key];
		});
		/**
		* The currently active debug mode names, and names to skip.
		*/
		createDebug.names = [];
		createDebug.skips = [];
		/**
		* Map of special "%n" handling functions, for the debug "format" argument.
		*
		* Valid key names are a single, lower or upper-case letter, i.e. "n" and "N".
		*/
		createDebug.formatters = {};
		/**
		* Selects a color for a debug namespace
		* @param {String} namespace The namespace string for the debug instance to be colored
		* @return {Number|String} An ANSI color code for the given namespace
		* @api private
		*/
		function selectColor(namespace) {
			let hash = 0;
			for (let i = 0; i < namespace.length; i++) {
				hash = (hash << 5) - hash + namespace.charCodeAt(i);
				hash |= 0;
			}
			return createDebug.colors[Math.abs(hash) % createDebug.colors.length];
		}
		createDebug.selectColor = selectColor;
		/**
		* Create a debugger with the given `namespace`.
		*
		* @param {String} namespace
		* @return {Function}
		* @api public
		*/
		function createDebug(namespace) {
			let prevTime;
			let enableOverride = null;
			let namespacesCache;
			let enabledCache;
			function debug(...args) {
				if (!debug.enabled) return;
				const self = debug;
				const curr = Number(/* @__PURE__ */ new Date());
				self.diff = curr - (prevTime || curr);
				self.prev = prevTime;
				self.curr = curr;
				prevTime = curr;
				args[0] = createDebug.coerce(args[0]);
				if (typeof args[0] !== "string") args.unshift("%O");
				let index = 0;
				args[0] = args[0].replace(/%([a-zA-Z%])/g, (match, format) => {
					if (match === "%%") return "%";
					index++;
					const formatter = createDebug.formatters[format];
					if (typeof formatter === "function") {
						const val = args[index];
						match = formatter.call(self, val);
						args.splice(index, 1);
						index--;
					}
					return match;
				});
				createDebug.formatArgs.call(self, args);
				(self.log || createDebug.log).apply(self, args);
			}
			debug.namespace = namespace;
			debug.useColors = createDebug.useColors();
			debug.color = createDebug.selectColor(namespace);
			debug.extend = extend;
			debug.destroy = createDebug.destroy;
			Object.defineProperty(debug, "enabled", {
				enumerable: true,
				configurable: false,
				get: () => {
					if (enableOverride !== null) return enableOverride;
					if (namespacesCache !== createDebug.namespaces) {
						namespacesCache = createDebug.namespaces;
						enabledCache = createDebug.enabled(namespace);
					}
					return enabledCache;
				},
				set: (v) => {
					enableOverride = v;
				}
			});
			if (typeof createDebug.init === "function") createDebug.init(debug);
			return debug;
		}
		function extend(namespace, delimiter) {
			const newDebug = createDebug(this.namespace + (typeof delimiter === "undefined" ? ":" : delimiter) + namespace);
			newDebug.log = this.log;
			return newDebug;
		}
		/**
		* Enables a debug mode by namespaces. This can include modes
		* separated by a colon and wildcards.
		*
		* @param {String} namespaces
		* @api public
		*/
		function enable(namespaces) {
			createDebug.save(namespaces);
			createDebug.namespaces = namespaces;
			createDebug.names = [];
			createDebug.skips = [];
			const split = (typeof namespaces === "string" ? namespaces : "").trim().replace(/\s+/g, ",").split(",").filter(Boolean);
			for (const ns of split) if (ns[0] === "-") createDebug.skips.push(ns.slice(1));
			else createDebug.names.push(ns);
		}
		/**
		* Checks if the given string matches a namespace template, honoring
		* asterisks as wildcards.
		*
		* @param {String} search
		* @param {String} template
		* @return {Boolean}
		*/
		function matchesTemplate(search, template) {
			let searchIndex = 0;
			let templateIndex = 0;
			let starIndex = -1;
			let matchIndex = 0;
			while (searchIndex < search.length) if (templateIndex < template.length && (template[templateIndex] === search[searchIndex] || template[templateIndex] === "*")) if (template[templateIndex] === "*") {
				starIndex = templateIndex;
				matchIndex = searchIndex;
				templateIndex++;
			} else {
				searchIndex++;
				templateIndex++;
			}
			else if (starIndex !== -1) {
				templateIndex = starIndex + 1;
				matchIndex++;
				searchIndex = matchIndex;
			} else return false;
			while (templateIndex < template.length && template[templateIndex] === "*") templateIndex++;
			return templateIndex === template.length;
		}
		/**
		* Disable debug output.
		*
		* @return {String} namespaces
		* @api public
		*/
		function disable() {
			const namespaces = [...createDebug.names, ...createDebug.skips.map((namespace) => "-" + namespace)].join(",");
			createDebug.enable("");
			return namespaces;
		}
		/**
		* Returns true if the given mode name is enabled, false otherwise.
		*
		* @param {String} name
		* @return {Boolean}
		* @api public
		*/
		function enabled(name) {
			for (const skip of createDebug.skips) if (matchesTemplate(name, skip)) return false;
			for (const ns of createDebug.names) if (matchesTemplate(name, ns)) return true;
			return false;
		}
		/**
		* Coerce `val`.
		*
		* @param {Mixed} val
		* @return {Mixed}
		* @api private
		*/
		function coerce(val) {
			if (val instanceof Error) return val.stack || val.message;
			return val;
		}
		/**
		* XXX DO NOT USE. This is a temporary stub function.
		* XXX It WILL be removed in the next major release.
		*/
		function destroy() {
			console.warn("Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`.");
		}
		createDebug.enable(createDebug.load());
		return createDebug;
	}
	module.exports = setup;
}));

//#endregion
//#region ../../node_modules/.pnpm/debug@4.4.3/node_modules/debug/src/browser.js
var require_browser = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* This is the web browser implementation of `debug()`.
	*/
	exports.formatArgs = formatArgs;
	exports.save = save;
	exports.load = load;
	exports.useColors = useColors;
	exports.storage = localstorage();
	exports.destroy = (() => {
		let warned = false;
		return () => {
			if (!warned) {
				warned = true;
				console.warn("Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`.");
			}
		};
	})();
	/**
	* Colors.
	*/
	exports.colors = [
		"#0000CC",
		"#0000FF",
		"#0033CC",
		"#0033FF",
		"#0066CC",
		"#0066FF",
		"#0099CC",
		"#0099FF",
		"#00CC00",
		"#00CC33",
		"#00CC66",
		"#00CC99",
		"#00CCCC",
		"#00CCFF",
		"#3300CC",
		"#3300FF",
		"#3333CC",
		"#3333FF",
		"#3366CC",
		"#3366FF",
		"#3399CC",
		"#3399FF",
		"#33CC00",
		"#33CC33",
		"#33CC66",
		"#33CC99",
		"#33CCCC",
		"#33CCFF",
		"#6600CC",
		"#6600FF",
		"#6633CC",
		"#6633FF",
		"#66CC00",
		"#66CC33",
		"#9900CC",
		"#9900FF",
		"#9933CC",
		"#9933FF",
		"#99CC00",
		"#99CC33",
		"#CC0000",
		"#CC0033",
		"#CC0066",
		"#CC0099",
		"#CC00CC",
		"#CC00FF",
		"#CC3300",
		"#CC3333",
		"#CC3366",
		"#CC3399",
		"#CC33CC",
		"#CC33FF",
		"#CC6600",
		"#CC6633",
		"#CC9900",
		"#CC9933",
		"#CCCC00",
		"#CCCC33",
		"#FF0000",
		"#FF0033",
		"#FF0066",
		"#FF0099",
		"#FF00CC",
		"#FF00FF",
		"#FF3300",
		"#FF3333",
		"#FF3366",
		"#FF3399",
		"#FF33CC",
		"#FF33FF",
		"#FF6600",
		"#FF6633",
		"#FF9900",
		"#FF9933",
		"#FFCC00",
		"#FFCC33"
	];
	/**
	* Currently only WebKit-based Web Inspectors, Firefox >= v31,
	* and the Firebug extension (any Firefox version) are known
	* to support "%c" CSS customizations.
	*
	* TODO: add a `localStorage` variable to explicitly enable/disable colors
	*/
	function useColors() {
		if (typeof window !== "undefined" && window.process && (window.process.type === "renderer" || window.process.__nwjs)) return true;
		if (typeof navigator !== "undefined" && navigator.userAgent && navigator.userAgent.toLowerCase().match(/(edge|trident)\/(\d+)/)) return false;
		let m;
		return typeof document !== "undefined" && document.documentElement && document.documentElement.style && document.documentElement.style.WebkitAppearance || typeof window !== "undefined" && window.console && (window.console.firebug || window.console.exception && window.console.table) || typeof navigator !== "undefined" && navigator.userAgent && (m = navigator.userAgent.toLowerCase().match(/firefox\/(\d+)/)) && parseInt(m[1], 10) >= 31 || typeof navigator !== "undefined" && navigator.userAgent && navigator.userAgent.toLowerCase().match(/applewebkit\/(\d+)/);
	}
	/**
	* Colorize log arguments if enabled.
	*
	* @api public
	*/
	function formatArgs(args) {
		args[0] = (this.useColors ? "%c" : "") + this.namespace + (this.useColors ? " %c" : " ") + args[0] + (this.useColors ? "%c " : " ") + "+" + module.exports.humanize(this.diff);
		if (!this.useColors) return;
		const c = "color: " + this.color;
		args.splice(1, 0, c, "color: inherit");
		let index = 0;
		let lastC = 0;
		args[0].replace(/%[a-zA-Z%]/g, (match) => {
			if (match === "%%") return;
			index++;
			if (match === "%c") lastC = index;
		});
		args.splice(lastC, 0, c);
	}
	/**
	* Invokes `console.debug()` when available.
	* No-op when `console.debug` is not a "function".
	* If `console.debug` is not available, falls back
	* to `console.log`.
	*
	* @api public
	*/
	exports.log = console.debug || console.log || (() => {});
	/**
	* Save `namespaces`.
	*
	* @param {String} namespaces
	* @api private
	*/
	function save(namespaces) {
		try {
			if (namespaces) exports.storage.setItem("debug", namespaces);
			else exports.storage.removeItem("debug");
		} catch (error) {}
	}
	/**
	* Load `namespaces`.
	*
	* @return {String} returns the previously persisted debug modes
	* @api private
	*/
	function load() {
		let r;
		try {
			r = exports.storage.getItem("debug") || exports.storage.getItem("DEBUG");
		} catch (error) {}
		if (!r && typeof process !== "undefined" && "env" in process) r = process.env.DEBUG;
		return r;
	}
	/**
	* Localstorage attempts to return the localstorage.
	*
	* This is necessary because safari throws
	* when a user disables cookies/localstorage
	* and you attempt to access it.
	*
	* @return {LocalStorage}
	* @api private
	*/
	function localstorage() {
		try {
			return localStorage;
		} catch (error) {}
	}
	module.exports = require_common()(exports);
	const { formatters } = module.exports;
	/**
	* Map %j to `JSON.stringify()`, since no Web Inspectors do that by default.
	*/
	formatters.j = function(v) {
		try {
			return JSON.stringify(v);
		} catch (error) {
			return "[UnexpectedJSONParseError]: " + error.message;
		}
	};
}));

//#endregion
//#region ../../node_modules/.pnpm/supports-color@10.2.2/node_modules/supports-color/index.js
var supports_color_exports = /* @__PURE__ */ __exportAll({
	createSupportsColor: () => createSupportsColor,
	default: () => supports_color_default
});
function hasFlag(flag, argv = globalThis.Deno ? globalThis.Deno.args : process$1.argv) {
	const prefix = flag.startsWith("-") ? "" : flag.length === 1 ? "-" : "--";
	const position = argv.indexOf(prefix + flag);
	const terminatorPosition = argv.indexOf("--");
	return position !== -1 && (terminatorPosition === -1 || position < terminatorPosition);
}
function envForceColor() {
	if (!("FORCE_COLOR" in env)) return;
	if (env.FORCE_COLOR === "true") return 1;
	if (env.FORCE_COLOR === "false") return 0;
	if (env.FORCE_COLOR.length === 0) return 1;
	const level = Math.min(Number.parseInt(env.FORCE_COLOR, 10), 3);
	if (![
		0,
		1,
		2,
		3
	].includes(level)) return;
	return level;
}
function translateLevel(level) {
	if (level === 0) return false;
	return {
		level,
		hasBasic: true,
		has256: level >= 2,
		has16m: level >= 3
	};
}
function _supportsColor(haveStream, { streamIsTTY, sniffFlags = true } = {}) {
	const noFlagForceColor = envForceColor();
	if (noFlagForceColor !== void 0) flagForceColor = noFlagForceColor;
	const forceColor = sniffFlags ? flagForceColor : noFlagForceColor;
	if (forceColor === 0) return 0;
	if (sniffFlags) {
		if (hasFlag("color=16m") || hasFlag("color=full") || hasFlag("color=truecolor")) return 3;
		if (hasFlag("color=256")) return 2;
	}
	if ("TF_BUILD" in env && "AGENT_NAME" in env) return 1;
	if (haveStream && !streamIsTTY && forceColor === void 0) return 0;
	const min = forceColor || 0;
	if (env.TERM === "dumb") return min;
	if (process$1.platform === "win32") {
		const osRelease = os.release().split(".");
		if (Number(osRelease[0]) >= 10 && Number(osRelease[2]) >= 10586) return Number(osRelease[2]) >= 14931 ? 3 : 2;
		return 1;
	}
	if ("CI" in env) {
		if ([
			"GITHUB_ACTIONS",
			"GITEA_ACTIONS",
			"CIRCLECI"
		].some((key) => key in env)) return 3;
		if ([
			"TRAVIS",
			"APPVEYOR",
			"GITLAB_CI",
			"BUILDKITE",
			"DRONE"
		].some((sign) => sign in env) || env.CI_NAME === "codeship") return 1;
		return min;
	}
	if ("TEAMCITY_VERSION" in env) return /^(9\.(0*[1-9]\d*)\.|\d{2,}\.)/.test(env.TEAMCITY_VERSION) ? 1 : 0;
	if (env.COLORTERM === "truecolor") return 3;
	if (env.TERM === "xterm-kitty") return 3;
	if (env.TERM === "xterm-ghostty") return 3;
	if (env.TERM === "wezterm") return 3;
	if ("TERM_PROGRAM" in env) {
		const version = Number.parseInt((env.TERM_PROGRAM_VERSION || "").split(".")[0], 10);
		switch (env.TERM_PROGRAM) {
			case "iTerm.app": return version >= 3 ? 3 : 2;
			case "Apple_Terminal": return 2;
		}
	}
	if (/-256(color)?$/i.test(env.TERM)) return 2;
	if (/^screen|^xterm|^vt100|^vt220|^rxvt|color|ansi|cygwin|linux/i.test(env.TERM)) return 1;
	if ("COLORTERM" in env) return 1;
	return min;
}
function createSupportsColor(stream, options = {}) {
	return translateLevel(_supportsColor(stream, {
		streamIsTTY: stream && stream.isTTY,
		...options
	}));
}
var env, flagForceColor, supportsColor, supports_color_default;
var init_supports_color = __esmMin((() => {
	({env} = process$1);
	;
	if (hasFlag("no-color") || hasFlag("no-colors") || hasFlag("color=false") || hasFlag("color=never")) flagForceColor = 0;
	else if (hasFlag("color") || hasFlag("colors") || hasFlag("color=true") || hasFlag("color=always")) flagForceColor = 1;
	supportsColor = {
		stdout: createSupportsColor({ isTTY: tty.isatty(1) }),
		stderr: createSupportsColor({ isTTY: tty.isatty(2) })
	};
	supports_color_default = supportsColor;
}));

//#endregion
//#region ../../node_modules/.pnpm/debug@4.4.3/node_modules/debug/src/node.js
var require_node = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* Module dependencies.
	*/
	const tty$1 = __require("tty");
	const util = __require("util");
	/**
	* This is the Node.js implementation of `debug()`.
	*/
	exports.init = init;
	exports.log = log;
	exports.formatArgs = formatArgs;
	exports.save = save;
	exports.load = load;
	exports.useColors = useColors;
	exports.destroy = util.deprecate(() => {}, "Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`.");
	/**
	* Colors.
	*/
	exports.colors = [
		6,
		2,
		3,
		4,
		5,
		1
	];
	try {
		const supportsColor = (init_supports_color(), __toCommonJS(supports_color_exports));
		if (supportsColor && (supportsColor.stderr || supportsColor).level >= 2) exports.colors = [
			20,
			21,
			26,
			27,
			32,
			33,
			38,
			39,
			40,
			41,
			42,
			43,
			44,
			45,
			56,
			57,
			62,
			63,
			68,
			69,
			74,
			75,
			76,
			77,
			78,
			79,
			80,
			81,
			92,
			93,
			98,
			99,
			112,
			113,
			128,
			129,
			134,
			135,
			148,
			149,
			160,
			161,
			162,
			163,
			164,
			165,
			166,
			167,
			168,
			169,
			170,
			171,
			172,
			173,
			178,
			179,
			184,
			185,
			196,
			197,
			198,
			199,
			200,
			201,
			202,
			203,
			204,
			205,
			206,
			207,
			208,
			209,
			214,
			215,
			220,
			221
		];
	} catch (error) {}
	/**
	* Build up the default `inspectOpts` object from the environment variables.
	*
	*   $ DEBUG_COLORS=no DEBUG_DEPTH=10 DEBUG_SHOW_HIDDEN=enabled node script.js
	*/
	exports.inspectOpts = Object.keys(process.env).filter((key) => {
		return /^debug_/i.test(key);
	}).reduce((obj, key) => {
		const prop = key.substring(6).toLowerCase().replace(/_([a-z])/g, (_, k) => {
			return k.toUpperCase();
		});
		let val = process.env[key];
		if (/^(yes|on|true|enabled)$/i.test(val)) val = true;
		else if (/^(no|off|false|disabled)$/i.test(val)) val = false;
		else if (val === "null") val = null;
		else val = Number(val);
		obj[prop] = val;
		return obj;
	}, {});
	/**
	* Is stdout a TTY? Colored output is enabled when `true`.
	*/
	function useColors() {
		return "colors" in exports.inspectOpts ? Boolean(exports.inspectOpts.colors) : tty$1.isatty(process.stderr.fd);
	}
	/**
	* Adds ANSI color escape codes if enabled.
	*
	* @api public
	*/
	function formatArgs(args) {
		const { namespace: name, useColors } = this;
		if (useColors) {
			const c = this.color;
			const colorCode = "\x1B[3" + (c < 8 ? c : "8;5;" + c);
			const prefix = `  ${colorCode};1m${name} \u001B[0m`;
			args[0] = prefix + args[0].split("\n").join("\n" + prefix);
			args.push(colorCode + "m+" + module.exports.humanize(this.diff) + "\x1B[0m");
		} else args[0] = getDate() + name + " " + args[0];
	}
	function getDate() {
		if (exports.inspectOpts.hideDate) return "";
		return (/* @__PURE__ */ new Date()).toISOString() + " ";
	}
	/**
	* Invokes `util.formatWithOptions()` with the specified arguments and writes to stderr.
	*/
	function log(...args) {
		return process.stderr.write(util.formatWithOptions(exports.inspectOpts, ...args) + "\n");
	}
	/**
	* Save `namespaces`.
	*
	* @param {String} namespaces
	* @api private
	*/
	function save(namespaces) {
		if (namespaces) process.env.DEBUG = namespaces;
		else delete process.env.DEBUG;
	}
	/**
	* Load `namespaces`.
	*
	* @return {String} returns the previously persisted debug modes
	* @api private
	*/
	function load() {
		return process.env.DEBUG;
	}
	/**
	* Init logic for `debug` instances.
	*
	* Create a new `inspectOpts` object in case `useColors` is set
	* differently for a particular `debug` instance.
	*/
	function init(debug) {
		debug.inspectOpts = {};
		const keys = Object.keys(exports.inspectOpts);
		for (let i = 0; i < keys.length; i++) debug.inspectOpts[keys[i]] = exports.inspectOpts[keys[i]];
	}
	module.exports = require_common()(exports);
	const { formatters } = module.exports;
	/**
	* Map %o to `util.inspect()`, all on a single line.
	*/
	formatters.o = function(v) {
		this.inspectOpts.colors = this.useColors;
		return util.inspect(v, this.inspectOpts).split("\n").map((str) => str.trim()).join(" ");
	};
	/**
	* Map %O to `util.inspect()`, allowing multiple lines if needed.
	*/
	formatters.O = function(v) {
		this.inspectOpts.colors = this.useColors;
		return util.inspect(v, this.inspectOpts);
	};
}));

//#endregion
//#region ../../node_modules/.pnpm/debug@4.4.3/node_modules/debug/src/index.js
var require_src = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* Detect Electron renderer / nwjs process, which is node, but we should
	* treat as a browser.
	*/
	if (typeof process === "undefined" || process.type === "renderer" || process.browser === true || process.__nwjs) module.exports = require_browser();
	else module.exports = require_node();
}));

//#endregion
//#region src/utils/logger.ts
var import_src = /* @__PURE__ */ __toESM(require_src(), 1);
const logger = R;
const debug = (0, import_src.default)("nuxi");

//#endregion
export { Ct$1 as _, Ie as a, Pt as c, Wt as d, Xt as f, zt as g, qt as h, supports_color_default as i, Qt as l, kt as m, logger as n, Mt as o, bt as p, init_supports_color as r, Nt as s, debug as t, Tt as u };
