// nuxi@3.33.0/dist/nypm-D5A4-HdH.mjs
import { t as __exportAll } from "./rolldown-runtime-CwrU2b7t.mjs";
import { delimiter, dirname, join, normalize, resolve } from "node:path";
import { cwd } from "node:process";
import c from "node:readline";
import { existsSync } from "node:fs";
import { readFile } from "node:fs/promises";
import { spawn } from "node:child_process";
import { PassThrough } from "node:stream";
import { createRequire as createRequire$1 } from "module";

//#region ../../node_modules/.pnpm/giget@3.1.1/node_modules/giget/dist/_chunks/libs/nypm.mjs
const _DRIVE_LETTER_START_RE = /^[A-Za-z]:\//;
function normalizeWindowsPath(e = ``) {
	return e && e.replace(/\\/g, `/`).replace(_DRIVE_LETTER_START_RE, (e) => e.toUpperCase());
}
const _UNC_REGEX = /^[/\\]{2}/, _IS_ABSOLUTE_RE = /^[/\\](?![/\\])|^[/\\]{2}(?!\.)|^[A-Za-z]:[/\\]/, _DRIVE_LETTER_RE = /^[A-Za-z]:$/, _ROOT_FOLDER_RE = /^\/([A-Za-z]:)?$/, normalize$1 = function(e) {
	if (e.length === 0) return `.`;
	e = normalizeWindowsPath(e);
	let r = e.match(_UNC_REGEX), i = isAbsolute$1(e), a = e[e.length - 1] === `/`;
	return e = normalizeString(e, !i), e.length === 0 ? i ? `/` : a ? `./` : `.` : (a && (e += `/`), _DRIVE_LETTER_RE.test(e) && (e += `/`), r ? i ? `//${e}` : `//./${e}` : i && !isAbsolute$1(e) ? `/${e}` : e);
}, join$1 = function(...e) {
	let r = ``;
	for (let i of e) if (i) if (r.length > 0) {
		let e = r[r.length - 1] === `/`, a = i[0] === `/`;
		e && a ? r += i.slice(1) : r += e || a ? i : `/${i}`;
	} else r += i;
	return normalize$1(r);
};
function cwd$1() {
	return typeof process < `u` && typeof process.cwd == `function` ? process.cwd().replace(/\\/g, `/`) : `/`;
}
const resolve$1 = function(...e) {
	e = e.map((e) => normalizeWindowsPath(e));
	let r = ``, i = !1;
	for (let a = e.length - 1; a >= -1 && !i; a--) {
		let o = a >= 0 ? e[a] : cwd$1();
		!o || o.length === 0 || (r = `${o}/${r}`, i = isAbsolute$1(o));
	}
	return r = normalizeString(r, !i), i && !isAbsolute$1(r) ? `/${r}` : r.length > 0 ? r : `.`;
};
function normalizeString(e, r) {
	let i = ``, a = 0, o = -1, s = 0, L = null;
	for (let J = 0; J <= e.length; ++J) {
		if (J < e.length) L = e[J];
		else if (L === `/`) break;
		else L = `/`;
		if (L === `/`) {
			if (!(o === J - 1 || s === 1)) if (s === 2) {
				if (i.length < 2 || a !== 2 || i[i.length - 1] !== `.` || i[i.length - 2] !== `.`) {
					if (i.length > 2) {
						let e = i.lastIndexOf(`/`);
						e === -1 ? (i = ``, a = 0) : (i = i.slice(0, e), a = i.length - 1 - i.lastIndexOf(`/`)), o = J, s = 0;
						continue;
					} else if (i.length > 0) {
						i = ``, a = 0, o = J, s = 0;
						continue;
					}
				}
				r && (i += i.length > 0 ? `/..` : `..`, a = 2);
			} else i.length > 0 ? i += `/${e.slice(o + 1, J)}` : i = e.slice(o + 1, J), a = J - o - 1;
			o = J, s = 0;
		} else L === `.` && s !== -1 ? ++s : s = -1;
	}
	return i;
}
const isAbsolute$1 = function(e) {
	return _IS_ABSOLUTE_RE.test(e);
}, relative = function(e, r) {
	let i = resolve$1(e).replace(_ROOT_FOLDER_RE, `$1`).split(`/`), a = resolve$1(r).replace(_ROOT_FOLDER_RE, `$1`).split(`/`);
	if (a[0][1] === `:` && i[0][1] === `:` && i[0] !== a[0]) return a.join(`/`);
	let o = [...i];
	for (let e of o) {
		if (a[0] !== e) break;
		i.shift(), a.shift();
	}
	return [...i.map(() => `..`), ...a].join(`/`);
}, dirname$1 = function(e) {
	let r = normalizeWindowsPath(e).replace(/\/$/, ``).split(`/`).slice(0, -1);
	return r.length === 1 && _DRIVE_LETTER_RE.test(r[0]) && (r[0] += `/`), r.join(`/`) || (isAbsolute$1(e) ? `/` : `.`);
}, basename$1 = function(e, r) {
	let i = normalizeWindowsPath(e).split(`/`), a = ``;
	for (let e = i.length - 1; e >= 0; e--) {
		let r = i[e];
		if (r) {
			a = r;
			break;
		}
	}
	return r && a.endsWith(r) ? a.slice(0, -r.length) : a;
};
var l = Object.create, u = Object.defineProperty, d = Object.getOwnPropertyDescriptor, f = Object.getOwnPropertyNames, p = Object.getPrototypeOf, m = Object.prototype.hasOwnProperty, h = (e, r) => () => (r || e((r = { exports: {} }).exports, r), r.exports), g = (e, r, i, a) => {
	if (r && typeof r == `object` || typeof r == `function`) for (var o = f(r), s = 0, L = o.length, J; s < L; s++) J = o[s], !m.call(e, J) && J !== i && u(e, J, {
		get: ((e) => r[e]).bind(null, J),
		enumerable: !(a = d(r, J)) || a.enumerable
	});
	return e;
}, _ = (e, r, i) => (i = e == null ? {} : l(p(e)), g(r || !e || !e.__esModule ? u(i, `default`, {
	value: e,
	enumerable: !0
}) : i, e)), v = createRequire$1(import.meta.url);
const y = /^path$/i, b = {
	key: `PATH`,
	value: ``
};
function x(e) {
	for (let r in e) {
		if (!Object.prototype.hasOwnProperty.call(e, r) || !y.test(r)) continue;
		let i = e[r];
		return i ? {
			key: r,
			value: i
		} : b;
	}
	return b;
}
function S(e, r) {
	let i = r.value.split(delimiter), a = e, o;
	do
		i.push(resolve(a, `node_modules`, `.bin`)), o = a, a = dirname(a);
	while (a !== o);
	return {
		key: r.key,
		value: i.join(delimiter)
	};
}
function C(e, r) {
	let i = {
		...process.env,
		...r
	}, a = S(e, x(i));
	return i[a.key] = a.value, i;
}
const w = (e) => {
	let r = e.length, i = new PassThrough(), o = () => {
		--r === 0 && i.emit(`end`);
	};
	for (let r of e) r.pipe(i, { end: !1 }), r.on(`end`, o);
	return i;
};
var T = h((e, r) => {
	r.exports = s, s.sync = L;
	var i = v(`fs`);
	function a(e, r) {
		var i = r.pathExt === void 0 ? process.env.PATHEXT : r.pathExt;
		if (!i || (i = i.split(`;`), i.indexOf(``) !== -1)) return !0;
		for (var a = 0; a < i.length; a++) {
			var o = i[a].toLowerCase();
			if (o && e.substr(-o.length).toLowerCase() === o) return !0;
		}
		return !1;
	}
	function o(e, r, i) {
		return !e.isSymbolicLink() && !e.isFile() ? !1 : a(r, i);
	}
	function s(e, r, a) {
		i.stat(e, function(i, s) {
			a(i, i ? !1 : o(s, e, r));
		});
	}
	function L(e, r) {
		return o(i.statSync(e), e, r);
	}
}), E = h((e, r) => {
	r.exports = a, a.sync = o;
	var i = v(`fs`);
	function a(e, r, a) {
		i.stat(e, function(e, i) {
			a(e, e ? !1 : s(i, r));
		});
	}
	function o(e, r) {
		return s(i.statSync(e), r);
	}
	function s(e, r) {
		return e.isFile() && L(e, r);
	}
	function L(e, r) {
		var i = e.mode, a = e.uid, o = e.gid, s = r.uid === void 0 ? process.getuid && process.getuid() : r.uid, L = r.gid === void 0 ? process.getgid && process.getgid() : r.gid, J = 64, Y = 8, X = 1, Z = J | Y;
		return i & X || i & Y && o === L || i & J && a === s || i & Z && s === 0;
	}
}), D = h((e, r) => {
	v(`fs`);
	var i = process.platform === `win32` || global.TESTING_WINDOWS ? T() : E();
	r.exports = a, a.sync = o;
	function a(e, r, o) {
		if (typeof r == `function` && (o = r, r = {}), !o) {
			if (typeof Promise != `function`) throw TypeError(`callback not provided`);
			return new Promise(function(i, o) {
				a(e, r || {}, function(e, r) {
					e ? o(e) : i(r);
				});
			});
		}
		i(e, r || {}, function(e, i) {
			e && (e.code === `EACCES` || r && r.ignoreErrors) && (e = null, i = !1), o(e, i);
		});
	}
	function o(e, r) {
		try {
			return i.sync(e, r || {});
		} catch (e) {
			if (r && r.ignoreErrors || e.code === `EACCES`) return !1;
			throw e;
		}
	}
}), O = h((e, r) => {
	let i = process.platform === `win32` || process.env.OSTYPE === `cygwin` || process.env.OSTYPE === `msys`, a = v(`path`), o = i ? `;` : `:`, s = D(), L = (e) => Object.assign(Error(`not found: ${e}`), { code: `ENOENT` }), J = (e, r) => {
		let a = r.colon || o, s = e.match(/\//) || i && e.match(/\\/) ? [``] : [...i ? [process.cwd()] : [], ...(r.path || process.env.PATH || ``).split(a)], L = i ? r.pathExt || process.env.PATHEXT || `.EXE;.CMD;.BAT;.COM` : ``, J = i ? L.split(a) : [``];
		return i && e.indexOf(`.`) !== -1 && J[0] !== `` && J.unshift(``), {
			pathEnv: s,
			pathExt: J,
			pathExtExe: L
		};
	}, Y = (e, r, i) => {
		typeof r == `function` && (i = r, r = {}), r ||= {};
		let { pathEnv: o, pathExt: Y, pathExtExe: X } = J(e, r), Z = [], Q = (i) => new Promise((s, J) => {
			if (i === o.length) return r.all && Z.length ? s(Z) : J(L(e));
			let Y = o[i], X = /^".*"$/.test(Y) ? Y.slice(1, -1) : Y, Q = a.join(X, e);
			s($(!X && /^\.[\\\/]/.test(e) ? e.slice(0, 2) + Q : Q, i, 0));
		}), $ = (e, i, a) => new Promise((o, L) => {
			if (a === Y.length) return o(Q(i + 1));
			let J = Y[a];
			s(e + J, { pathExt: X }, (s, L) => {
				if (!s && L) if (r.all) Z.push(e + J);
				else return o(e + J);
				return o($(e, i, a + 1));
			});
		});
		return i ? Q(0).then((e) => i(null, e), i) : Q(0);
	};
	r.exports = Y, Y.sync = (e, r) => {
		r ||= {};
		let { pathEnv: i, pathExt: o, pathExtExe: Y } = J(e, r), X = [];
		for (let L = 0; L < i.length; L++) {
			let J = i[L], Z = /^".*"$/.test(J) ? J.slice(1, -1) : J, Q = a.join(Z, e), $ = !Z && /^\.[\\\/]/.test(e) ? e.slice(0, 2) + Q : Q;
			for (let e = 0; e < o.length; e++) {
				let i = $ + o[e];
				try {
					if (s.sync(i, { pathExt: Y })) if (r.all) X.push(i);
					else return i;
				} catch {}
			}
		}
		if (r.all && X.length) return X;
		if (r.nothrow) return null;
		throw L(e);
	};
}), k = h((e, r) => {
	let i = (e = {}) => {
		let r = e.env || process.env;
		return (e.platform || process.platform) === `win32` ? Object.keys(r).reverse().find((e) => e.toUpperCase() === `PATH`) || `Path` : `PATH`;
	};
	r.exports = i, r.exports.default = i;
}), A = h((e, r) => {
	let i = v(`path`), a = O(), o = k();
	function s(e, r) {
		let s = e.options.env || process.env, L = process.cwd(), J = e.options.cwd != null, Y = J && process.chdir !== void 0 && !process.chdir.disabled;
		if (Y) try {
			process.chdir(e.options.cwd);
		} catch {}
		let X;
		try {
			X = a.sync(e.command, {
				path: s[o({ env: s })],
				pathExt: r ? i.delimiter : void 0
			});
		} catch {} finally {
			Y && process.chdir(L);
		}
		return X &&= i.resolve(J ? e.options.cwd : ``, X), X;
	}
	function L(e) {
		return s(e) || s(e, !0);
	}
	r.exports = L;
}), j = h((e, r) => {
	let i = /([()\][%!^"`<>&|;, *?])/g;
	function a(e) {
		return e = e.replace(i, `^$1`), e;
	}
	function o(e, r) {
		return e = `${e}`, e = e.replace(/(\\*)"/g, `$1$1\\"`), e = e.replace(/(\\*)$/, `$1$1`), e = `"${e}"`, e = e.replace(i, `^$1`), r && (e = e.replace(i, `^$1`)), e;
	}
	r.exports.command = a, r.exports.argument = o;
}), M = h((e, r) => {
	r.exports = /^#!(.*)/;
}), N = h((e, r) => {
	let i = M();
	r.exports = (e = ``) => {
		let r = e.match(i);
		if (!r) return null;
		let [a, o] = r[0].replace(/#! ?/, ``).split(` `), s = a.split(`/`).pop();
		return s === `env` ? o : o ? `${s} ${o}` : s;
	};
}), P = h((e, r) => {
	let i = v(`fs`), a = N();
	function o(e) {
		let r = Buffer.alloc(150), o;
		try {
			o = i.openSync(e, `r`), i.readSync(o, r, 0, 150, 0), i.closeSync(o);
		} catch {}
		return a(r.toString());
	}
	r.exports = o;
}), F = h((e, r) => {
	let i = v(`path`), a = A(), o = j(), s = P(), L = process.platform === `win32`, J = /\.(?:com|exe)$/i, Y = /node_modules[\\/].bin[\\/][^\\/]+\.cmd$/i;
	function X(e) {
		e.file = a(e);
		let r = e.file && s(e.file);
		return r ? (e.args.unshift(e.file), e.command = r, a(e)) : e.file;
	}
	function Z(e) {
		if (!L) return e;
		let r = X(e), a = !J.test(r);
		if (e.options.forceShell || a) {
			let a = Y.test(r);
			e.command = i.normalize(e.command), e.command = o.command(e.command), e.args = e.args.map((e) => o.argument(e, a)), e.args = [
				`/d`,
				`/s`,
				`/c`,
				`"${[e.command].concat(e.args).join(` `)}"`
			], e.command = process.env.comspec || `cmd.exe`, e.options.windowsVerbatimArguments = !0;
		}
		return e;
	}
	function Q(e, r, i) {
		r && !Array.isArray(r) && (i = r, r = null), r = r ? r.slice(0) : [], i = Object.assign({}, i);
		let a = {
			command: e,
			args: r,
			options: i,
			file: void 0,
			original: {
				command: e,
				args: r
			}
		};
		return i.shell ? a : Z(a);
	}
	r.exports = Q;
}), I = h((e, r) => {
	let i = process.platform === `win32`;
	function a(e, r) {
		return Object.assign(Error(`${r} ${e.command} ENOENT`), {
			code: `ENOENT`,
			errno: `ENOENT`,
			syscall: `${r} ${e.command}`,
			path: e.command,
			spawnargs: e.args
		});
	}
	function o(e, r) {
		if (!i) return;
		let a = e.emit;
		e.emit = function(i, o) {
			if (i === `exit`) {
				let i = s(o, r, `spawn`);
				if (i) return a.call(e, `error`, i);
			}
			return a.apply(e, arguments);
		};
	}
	function s(e, r) {
		return i && e === 1 && !r.file ? a(r.original, `spawn`) : null;
	}
	function L(e, r) {
		return i && e === 1 && !r.file ? a(r.original, `spawnSync`) : null;
	}
	r.exports = {
		hookChildProcess: o,
		verifyENOENT: s,
		verifyENOENTSync: L,
		notFoundError: a
	};
}), R = _(h((e, r) => {
	let i = v(`child_process`), a = F(), o = I();
	function s(e, r, s) {
		let L = a(e, r, s), J = i.spawn(L.command, L.args, L.options);
		return o.hookChildProcess(J, L), J;
	}
	function L(e, r, s) {
		let L = a(e, r, s), J = i.spawnSync(L.command, L.args, L.options);
		return J.error = J.error || o.verifyENOENTSync(J.status, L), J;
	}
	r.exports = s, r.exports.spawn = s, r.exports.sync = L, r.exports._parse = a, r.exports._enoent = o;
})(), 1), z = class extends Error {
	result;
	output;
	get exitCode() {
		if (this.result.exitCode !== null) return this.result.exitCode;
	}
	constructor(e, r) {
		super(`Process exited with non-zero status (${e.exitCode})`), this.result = e, this.output = r;
	}
};
const B = {
	timeout: void 0,
	persist: !1
}, V = { windowsHide: !0 };
function H(e, r) {
	return {
		command: normalize(e),
		args: r ?? []
	};
}
function U(e) {
	let r = new AbortController();
	for (let i of e) {
		if (i.aborted) return r.abort(), i;
		i.addEventListener(`abort`, () => {
			r.abort(i.reason);
		}, { signal: r.signal });
	}
	return r.signal;
}
async function W(e) {
	let r = ``;
	for await (let i of e) r += i.toString();
	return r;
}
var G = class {
	_process;
	_aborted = !1;
	_options;
	_command;
	_args;
	_resolveClose;
	_processClosed;
	_thrownError;
	get process() {
		return this._process;
	}
	get pid() {
		return this._process?.pid;
	}
	get exitCode() {
		if (this._process && this._process.exitCode !== null) return this._process.exitCode;
	}
	constructor(e, r, i) {
		this._options = {
			...B,
			...i
		}, this._command = e, this._args = r ?? [], this._processClosed = new Promise((e) => {
			this._resolveClose = e;
		});
	}
	kill(e) {
		return this._process?.kill(e) === !0;
	}
	get aborted() {
		return this._aborted;
	}
	get killed() {
		return this._process?.killed === !0;
	}
	pipe(e, r, i) {
		return q(e, r, {
			...i,
			stdin: this
		});
	}
	async *[Symbol.asyncIterator]() {
		let e = this._process;
		if (!e) return;
		let r = [];
		this._streamErr && r.push(this._streamErr), this._streamOut && r.push(this._streamOut);
		let i = w(r), a = c.createInterface({ input: i });
		for await (let e of a) yield e.toString();
		if (await this._processClosed, e.removeAllListeners(), this._thrownError) throw this._thrownError;
		if (this._options?.throwOnError && this.exitCode !== 0 && this.exitCode !== void 0) throw new z(this);
	}
	async _waitForOutput() {
		let e = this._process;
		if (!e) throw Error(`No process was started`);
		let [r, i] = await Promise.all([this._streamOut ? W(this._streamOut) : ``, this._streamErr ? W(this._streamErr) : ``]);
		if (await this._processClosed, this._options?.stdin && await this._options.stdin, e.removeAllListeners(), this._thrownError) throw this._thrownError;
		let a = {
			stderr: i,
			stdout: r,
			exitCode: this.exitCode
		};
		if (this._options.throwOnError && this.exitCode !== 0 && this.exitCode !== void 0) throw new z(this, a);
		return a;
	}
	then(e, r) {
		return this._waitForOutput().then(e, r);
	}
	_streamOut;
	_streamErr;
	spawn() {
		let e = cwd(), r = this._options, i = {
			...V,
			...r.nodeOptions
		}, a = [];
		this._resetState(), r.timeout !== void 0 && a.push(AbortSignal.timeout(r.timeout)), r.signal !== void 0 && a.push(r.signal), r.persist === !0 && (i.detached = !0), a.length > 0 && (i.signal = U(a)), i.env = C(e, i.env);
		let { command: s, args: L } = H(this._command, this._args), J = (0, R._parse)(s, L, i), Y = spawn(J.command, J.args, J.options);
		if (Y.stderr && (this._streamErr = Y.stderr), Y.stdout && (this._streamOut = Y.stdout), this._process = Y, Y.once(`error`, this._onError), Y.once(`close`, this._onClose), r.stdin !== void 0 && Y.stdin && r.stdin.process) {
			let { stdout: e } = r.stdin.process;
			e && e.pipe(Y.stdin);
		}
	}
	_resetState() {
		this._aborted = !1, this._processClosed = new Promise((e) => {
			this._resolveClose = e;
		}), this._thrownError = void 0;
	}
	_onError = (e) => {
		if (e.name === `AbortError` && (!(e.cause instanceof Error) || e.cause.name !== `TimeoutError`)) {
			this._aborted = !0;
			return;
		}
		this._thrownError = e;
	};
	_onClose = () => {
		this._resolveClose && this._resolveClose();
	};
};
const K = (e, r, i) => {
	let a = new G(e, r, i);
	return a.spawn(), a;
}, q = K;
var dist_exports = __exportAll({
	detectPackageManager: () => detectPackageManager,
	installDependencies: () => installDependencies,
	packageManagers: () => packageManagers
});
async function findup(e, r, i = {}) {
	let a = normalize$1(e).split(`/`);
	for (; a.length > 0;) {
		let e = await r(a.join(`/`) || `/`);
		if (e || !i.includeParentDirs) return e;
		a.pop();
	}
}
function cached(e) {
	let r;
	return () => (r === void 0 && (r = e().then((e) => (r = e, r))), r);
}
const hasCorepack = cached(async () => {
	if (globalThis.process?.versions?.webcontainer) return !1;
	try {
		let { exitCode: e } = await K(`corepack`, [`--version`]);
		return e === 0;
	} catch {
		return !1;
	}
});
async function executeCommand(e, r, i = {}) {
	let a = e !== `npm` && e !== `bun` && e !== `deno` && i.corepack !== !1 && await hasCorepack() ? [`corepack`, [e, ...r]] : [e, r], { exitCode: o, stdout: s, stderr: L } = await K(a[0], a[1], { nodeOptions: {
		cwd: resolve$1(i.cwd || process.cwd()),
		env: i.env,
		stdio: i.silent ? `pipe` : `inherit`
	} });
	if (o !== 0) throw Error(`\`${a.flat().join(` `)}\` failed.${i.silent ? [
		``,
		s,
		L
	].join(`
`) : ``}`);
}
async function resolveOperationOptions(e = {}) {
	let r = e.cwd || process.cwd(), i = {
		...process.env,
		...e.env
	}, a = (typeof e.packageManager == `string` ? packageManagers.find((r) => r.name === e.packageManager) : e.packageManager) || await detectPackageManager(e.cwd || process.cwd());
	if (!a) throw Error(`No package manager auto-detected.`);
	return {
		cwd: r,
		env: i,
		silent: e.silent ?? !1,
		packageManager: a,
		dev: e.dev ?? !1,
		workspace: e.workspace,
		global: e.global ?? !1,
		dry: e.dry ?? !1,
		corepack: e.corepack ?? !0
	};
}
function parsePackageManagerField(e) {
	let [r, i] = (e || ``).split(`@`), [a, o] = i?.split(`+`) || [];
	if (r && r !== `-` && /^(@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/.test(r)) return {
		name: r,
		version: a,
		buildMeta: o
	};
	let s = (r || ``).replace(/\W+/g, ``);
	return {
		name: s,
		version: a,
		buildMeta: o,
		warnings: [`Abnormal characters found in \`packageManager\` field, sanitizing from \`${r}\` to \`${s}\``]
	};
}
const packageManagers = [
	{
		name: `npm`,
		command: `npm`,
		lockFile: `package-lock.json`
	},
	{
		name: `pnpm`,
		command: `pnpm`,
		lockFile: `pnpm-lock.yaml`,
		files: [`pnpm-workspace.yaml`]
	},
	{
		name: `bun`,
		command: `bun`,
		lockFile: [`bun.lockb`, `bun.lock`]
	},
	{
		name: `yarn`,
		command: `yarn`,
		lockFile: `yarn.lock`,
		files: [`.yarnrc.yml`]
	},
	{
		name: `deno`,
		command: `deno`,
		lockFile: `deno.lock`,
		files: [`deno.json`]
	}
];
async function detectPackageManager(e, a = {}) {
	let o = await findup(resolve$1(e || `.`), async (e) => {
		if (!a.ignorePackageJSON) {
			let a = join$1(e, `package.json`);
			if (existsSync(a)) {
				let e = JSON.parse(await readFile(a, `utf8`));
				if (e?.packageManager) {
					let { name: r, version: i = `0.0.0`, buildMeta: a, warnings: o } = parsePackageManagerField(e.packageManager);
					if (r) {
						let e = i.split(`.`)[0], s = packageManagers.find((i) => i.name === r && i.majorVersion === e) || packageManagers.find((e) => e.name === r);
						return {
							name: r,
							command: r,
							version: i,
							majorVersion: e,
							buildMeta: a,
							warnings: o,
							files: s?.files,
							lockFile: s?.lockFile
						};
					}
				}
			}
			if (existsSync(join$1(e, `deno.json`))) return packageManagers.find((e) => e.name === `deno`);
		}
		if (!a.ignoreLockFile) {
			for (let r of packageManagers) if ([r.lockFile, r.files].flat().filter(Boolean).some((r) => existsSync(resolve$1(e, r)))) return { ...r };
		}
	}, { includeParentDirs: a.includeParentDirs ?? !0 });
	if (!o && !a.ignoreArgv) {
		let e = process.argv[1];
		if (e) {
			for (let r of packageManagers) if (RegExp(`[/\\\\]\\.?${r.command}`).test(e)) return r;
		}
	}
	return o;
}
async function installDependencies(e = {}) {
	let r = await resolveOperationOptions(e), i = e.frozenLockFile ? {
		npm: [`ci`],
		yarn: [`install`, `--immutable`],
		bun: [`install`, `--frozen-lockfile`],
		pnpm: [`install`, `--frozen-lockfile`],
		deno: [`install`, `--frozen`]
	}[r.packageManager.name] : [`install`];
	return e.ignoreWorkspace && r.packageManager.name === `pnpm` && i.push(`--ignore-workspace`), r.dry || await executeCommand(r.packageManager.command, i, {
		cwd: r.cwd,
		silent: r.silent,
		corepack: r.corepack
	}), { exec: {
		command: r.packageManager.command,
		args: i
	} };
}

//#endregion
export { resolve$1 as a, relative as i, dirname$1 as n, dist_exports as r, basename$1 as t };
